<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Dina Schuster" />

<meta name="date" content="2023-01-20" />

<title>Single Dose Treatment Data Analysis Workflow</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Single Dose Treatment Data Analysis
Workflow</h1>
<h4 class="author">Dina Schuster</h4>
<h4 class="date">2023-01-20</h4>



<style>
body {
text-align: justify}
</style>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This vignette will give you an overview of how you can analyse
bottom-up proteomics data or LiP-MS data using <strong>protti</strong>.
If you would like to analyse dose-response data please refer to the <a href="https://jpquast.github.io/protti/articles/data_analysis_dose_response_workflow.html">dose-response
data analysis vignette</a>. Before analysing your data make sure that it
is of sufficient quality and that you do not have any outliers. To do
this you can take a look at the <a href="https://jpquast.github.io/protti/articles/quality_control_workflow.html">quality
control vignette</a>.</p>
<p><strong>protti</strong> includes several functions that make it easy
for the user to analyse and interpret data from bottom-up proteomics or
LiP-MS experiments. The R package includes functions for</p>
<ul>
<li>Quality control (see <a href="https://jpquast.github.io/protti/articles/quality_control_workflow.html">quality
control vignette</a> for more detailed information)</li>
<li>Data preparation
<ul>
<li>Median normalisation</li>
<li>Data filtering</li>
<li>Protein abundance calculation from precursor or peptide
intensities</li>
<li>Imputation of missing values</li>
<li>Fetching of database information (ChEBI, GO, KEGG, MobiDB,
UniProt)</li>
<li>Calculation of sequence coverage</li>
</ul></li>
<li>Data analysis
<ul>
<li>Statistical hypothesis tests</li>
<li>Fitting of dose-response curves (relevant for experiments with
several treatment concentrations, click <a href="https://jpquast.github.io/protti/articles/data_analysis_dose_response_workflow.html">here</a>
for more information)</li>
</ul></li>
<li>Data visualisation
<ul>
<li>Volcano plots</li>
<li>Barcode plots (plots that show protein coverage and significant
changes projected onto the protein sequence)</li>
<li>Wood’s plots</li>
<li>Profile plots</li>
</ul></li>
<li>Data interpretation
<ul>
<li>Treatment enrichment (check if your hits are enriched with known
targets)</li>
<li>GO-term enrichment</li>
<li>Network analysis (based on STRING database information)</li>
<li>KEGG pathway enrichment</li>
</ul></li>
</ul>
<p>You can read more about specific functions and how to use them by
calling e.g. <code>?normalise</code> (for the <code>normalise()</code>
function). Calling <code>?</code> followed by the function name will
display the function documentation and give you more detailed
information about the function. This can be done for any of the
functions included in the package.</p>
<p>This document will give you an overview of data preparation, data
analysis, data visualisation and data interpretation functions included
in <strong>protti</strong>. It will show you how they can be applied to
your data. The examples in this file are run on a <a href="https://www.nature.com/articles/s41467-020-18071-x">published</a>
LiP-MS data set. In the experiment a HeLa cell lysate was treated with
different amounts of the drug rapamycin and the target of rapamycin
(FKBP12) was successfully identified. For this vignette we are using a
filtered version of the original data set where we include only the 10
μM treatment concentration and the control (untreated). For simplicity
only 50 proteins are included in the data set.</p>
<p>The data set is produced from the output of <a href="https://biognosys.com/software/spectronaut/">Spectronaut™</a>.
However, if you have any other data such as DDA data that was searched
with a different search engine you can still apply
<strong>protti</strong>’s functions. Just make sure that your data frame
contains <a href="https://r4ds.had.co.nz/tidy-data.html">tidy data</a>.
That means data should be contained in a long format (e.g. all sample
names in one column) rather than a wide format (e.g. each sample name in
its own column). You can easily achieve this by using the
<code>pivot_longer()</code> function from the <code>tidyr</code>
package. If you are unsure what your input data is supposed to look
like, please use the <code>create_synthetic_data()</code> function and
compare this to your data. You can also take a look at the <a href="https://jpquast.github.io/protti/articles/input_preparation_workflow.html">input
preparation vignette</a>, there you will find all the necessary
information on how to get your data into the correct format.</p>
<p>The input data should have a similar structure to this example:</p>
<table>
<thead>
<tr class="header">
<th align="center">Sample</th>
<th align="center">Replicate</th>
<th align="center">Peptide</th>
<th align="center">Condition</th>
<th align="center">log2(Intensity)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">sample1</td>
<td align="center">1</td>
<td align="center">PEPTIDER</td>
<td align="center">treated</td>
<td align="center">14</td>
</tr>
<tr class="even">
<td align="center">sample1</td>
<td align="center">1</td>
<td align="center">PEPTI</td>
<td align="center">treated</td>
<td align="center">16</td>
</tr>
<tr class="odd">
<td align="center">sample1</td>
<td align="center">1</td>
<td align="center">PEPTIDE</td>
<td align="center">treated</td>
<td align="center">17</td>
</tr>
<tr class="even">
<td align="center">sample2</td>
<td align="center">1</td>
<td align="center">PEPTIDER</td>
<td align="center">untreated</td>
<td align="center">15</td>
</tr>
<tr class="odd">
<td align="center">sample2</td>
<td align="center">1</td>
<td align="center">PEPTI</td>
<td align="center">untreated</td>
<td align="center">18</td>
</tr>
<tr class="even">
<td align="center">sample2</td>
<td align="center">1</td>
<td align="center">PEPTIDE</td>
<td align="center">untreated</td>
<td align="center">12</td>
</tr>
</tbody>
</table>
</div>
<div id="how-to-use-protti-to-analyse-your-data" class="section level1">
<h1>How to use protti to analyse your data</h1>
<div id="getting-started" class="section level2">
<h2>Getting started</h2>
<p>Before we can start analysing our data, we need to load the
<strong>protti</strong> package. This is done by using the base R
function <code>library()</code>. In addition, we are also loading the
packages <code>magrittr</code> and <code>dplyr</code>. Both
<code>magrittr</code> and <code>dplyr</code> are part of the <a href="https://www.tidyverse.org">tidyverse</a>, a collection of packages
that provide useful functionalities for data processing and
visualisation. If you use many tidyverse packages in your workflow you
can easily load all at once by calling
<code>library(tidyverse)</code>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(protti)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<p>After having loaded the required packages we will load our data set
into the R environment. In order to do this for your data set you can
use the function <code>read_protti()</code>. This function is a wrapper
around the fast <code>fread()</code> function from the <a href="https://rdatatable.gitlab.io/data.table/"><code>data.table</code></a>
package and the <code>clean_names()</code> function from the <a href="https://sfirke.github.io/janitor/"><code>janitor</code></a>
package. This will allow you to not only load your data into R very
fast, but also to clean up the column names into lower snake case to
make them more R-friendly. This will make it easier to remember them and
to use them in your data analysis.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># To read in your own data you can use read_protti()</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>your_data <span class="ot">&lt;-</span> <span class="fu">read_protti</span>(<span class="at">filename =</span> <span class="st">&quot;mydata/data.csv&quot;</span>)</span></code></pre></div>
<p>For this example we are going to use the <code>rapamycin_10uM</code>
test data set included in <strong>protti</strong>. To read in the file
we are simply going to use the <code>utils</code> function
<code>data()</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;rapamycin_10uM&quot;</span>)</span></code></pre></div>
</div>
<div id="data-preparation" class="section level2">
<h2>Data preparation</h2>
<div id="log2-transformation-median-normalisation-and-cv-filtering" class="section level3">
<h3>Log2 transformation, median normalisation and CV filtering</h3>
<p>After inspecting the data and performing quality control (see <a href="https://jpquast.github.io/protti/articles/quality_control_workflow.html">quality
control vignette</a> for more information) we will now start to prepare
the data for the analysis.</p>
<p>First, we remove decoy hits (used for false discovery rate
estimation). Our example data set contains a column called
<code>eg_is_decoy</code> that consists of logicals indicating whether or
not the peptide is a decoy hit. To remove decoys we will use the
<code>dplyr</code> function <code>filter()</code>.</p>
<p>Next, we log2 transform our intensities, then normalise the data to
the median value of all runs. To transform the intensities we use the
<code>dplyr</code> function <code>mutate()</code> which creates a new
column while maintaining the original column.</p>
<p>Note that we are also using the pipe operator <code>%&gt;%</code>
included in the R package <a href="https://magrittr.tidyverse.org/index.html"><code>magrittr</code></a>.
<code>%&gt;%</code> takes the output of the preceding function and
supplies it as the first argument of the following function. Using
<code>%&gt;%</code> makes code easier to read and follow.</p>
<p>For normalisation we are using the <strong>protti</strong> function
<code>normalise()</code>. For this example we will use median
normalisation (<code>method = &quot;median&quot;</code>). The function normalises
intensities for each run to the median of all runs. This is only
necessary if your search algorithm does not already median normalise
your intensities. For the example data we have disabled median
normalisation in Spectronaut, therefore we need to median normalise now.
The formula for median normalisation is:</p>
<div class="blackbox">
<p>
<span class="math display"><span class="math display">\[median ~
normalised ~ intensity  =
intensity - median ( run ~ intensity ) + median ( global ~ intensity)
\]</span></span>
</p>
</div>
<p>To ensure that only good peptide measurements will be used for
further analysis, we will also filter our data based on coefficients of
variation (CV). In order to do this we are using the function
<code>filter_cv()</code>. For this example we are retaining peptides
with a CV &lt; 25 % in at least one of the two conditions.</p>
<p>The CVs are calculated within the function according to the
formula:</p>
<div class="blackbox">
<p>
<span class="math display"><span class="math display">\[CV =
\frac{standard ~ deviation}{mean} *
100\]</span></span>
</p>
</div>
<p><em>Note: The use of the <code>filter_cv()</code> function is
optional. It might remove a lot of your data if your experiment was
noisy. However, especially in these cases, the function will remove
peptides with poor quality and should improve the result. It is very
important that if you use this function you should not use the moderated
t-test or proDA algorithm on your data for differential abundance
estimation and significance testing. This likely will lead to an
inflated false positive rate because it alters the distributional
assumptions of these tests (<a href="https://pubmed.ncbi.nlm.nih.gov/20460310/">Bourgon
2010</a>).</em></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>data_normalised <span class="ot">&lt;-</span> rapamycin_10uM <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(eg_is_decoy <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">intensity_log2 =</span> <span class="fu">log2</span>(fg_quantity)) <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">normalise</span>(<span class="at">sample =</span> r_file_name, </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">intensity_log2 =</span> intensity_log2,</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">method =</span> <span class="st">&quot;median&quot;</span>)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>data_filtered <span class="ot">&lt;-</span> data_normalised <span class="sc">%&gt;%</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter_cv</span>(<span class="at">grouping =</span> eg_precursor_id, </span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">condition =</span> r_condition, </span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">log2_intensity =</span> intensity_log2, </span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">cv_limit =</span> <span class="fl">0.25</span>,</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">min_conditions =</span> <span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="remove-non-proteotypic-peptides" class="section level3">
<h3>Remove non-proteotypic peptides</h3>
<p>For LiP-MS analysis we commonly remove non-proteotypic peptides
(i.e. peptides that could come from more than one protein). If you
detect a change in non-proteotypic peptides it is not possible to
clearly assign which protein it comes from and therefore which protein
is affected by your treatment. If you are using the output from
Spectronaut you will find a column called “pep_is_proteotypic”. This
column contains logicals indicating whether your peptide is proteotypic
or not.</p>
<p>To filter out non proteotypic peptides we are using the
<code>dplyr</code> function <code>filter()</code>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>data_filtered_proteotypic <span class="ot">&lt;-</span> data_filtered <span class="sc">%&gt;%</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pep_is_proteotypic <span class="sc">==</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="fetching-database-information-and-assigning-peptide-types" class="section level3">
<h3>Fetching database information and assigning peptide types</h3>
<p>In order to obtain more information about our identified proteins we
are going to use the function <code>fetch_uniprot()</code> to download
information from UniProt directly.</p>
<p><code>fetch_uniprot()</code> uses a vector of UniProt IDs as its
input. We produce this vector by using the base R function
<code>unique()</code> which will extract all unique elements in the
selected protein ID column. In this case we want to download the full
protein name, gene IDs, GO terms associated with molecular function,
StringDB IDs, information on known interacting proteins, location of the
active site, location of binding sites, PDB entries, protein length and
protein sequence. There are more options for columns to add (for more
information on possible columns to add click <a href="https://www.uniprot.org/help/return_fields">here</a>).</p>
<p><code>fetch_uniprot()</code> returns a new data frame. In order to be
able to merge this with our original data frame we have to rename the ID
column to match the name of the protein ID column of our original data
frame. To do this we use <code>dplyr</code>’s <code>rename()</code>
function.</p>
<p>To merge the two data frames we use the <code>dplyr</code> function
<code>left_join()</code>. We match the two data frames by the column
“pg_protein_accessions”. By using <code>left_join()</code> we retain all
rows from our original data frame while adding the columns from the data
fame generated with <code>fetch_uniprot()</code>.</p>
<p><em>Note: you can also directly join the UniProt data frame with your
data without the need to rename its <code>id</code> column. You can
specify in the <code>by</code> argument in <code>left_join()</code> that
two columns are differently named.</em></p>
<p>Next, we would like to assign the trypticity of our peptides (i.e. if
the peptides are fully-tryptic, semi-tryptic or non-tryptic). In order
to do this we first need to define the peptide positions in the protein
and find the preceding and following amino acids. To obtain this
information we use the <strong>protti</strong> function
<code>find_peptide()</code>. The output of this function can then be
used in the function <code>assign_peptide_type()</code> which will add
an additional column with the peptide trypticity information. By using
the function <code>calculate_sequence_coverage()</code> we add an
additional column to the data frame containing information on how much
of the protein sequence we identified in our experiment.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>uniprot_ids <span class="ot">&lt;-</span> <span class="fu">unique</span>(data_filtered_proteotypic<span class="sc">$</span>pg_protein_accessions)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>uniprot <span class="ot">&lt;-</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fetch_uniprot</span>(</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">uniprot_ids =</span> uniprot_ids,</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">columns =</span> <span class="fu">c</span>(</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;protein_name&quot;</span>,</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;gene_names&quot;</span>,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;go_f&quot;</span>,</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;xref_string&quot;</span>,</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;cc_interaction&quot;</span>,</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;ft_act_site&quot;</span>,</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;ft_binding&quot;</span>,</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;xref_pdb&quot;</span>,</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;length&quot;</span>,</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;sequence&quot;</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">pg_protein_accessions =</span> accession)</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>data_filtered_uniprot <span class="ot">&lt;-</span> data_filtered_proteotypic <span class="sc">%&gt;%</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="at">y =</span> uniprot,</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="st">&quot;pg_protein_accessions&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">find_peptide</span>(<span class="at">protein_sequence =</span> sequence,</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">peptide_sequence =</span> pep_stripped_sequence) <span class="sc">%&gt;%</span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign_peptide_type</span>(<span class="at">aa_before =</span> aa_before,</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">last_aa =</span> last_aa, </span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>               <span class="at">aa_after =</span> aa_after) <span class="sc">%&gt;%</span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">calculate_sequence_coverage</span>(<span class="at">protein_sequence =</span> sequence,</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>                    <span class="at">peptides =</span> pep_stripped_sequence)</span></code></pre></div>
<p>With the <code>qc_sequence_coverage()</code> function, you check how
sequence coverage is distributed over all proteins in the sample.
Usually, the center of the distribution is low due to many proteins with
poor coverage. For this small data set with only 40 proteins the
sequence coverage is distributed relatively evenly.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qc_sequence_coverage</span>(</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_filtered_uniprot,</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">protein_identifier =</span> pg_protein_accessions,</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">coverage =</span> coverage</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
</div>
<div id="data-analysis" class="section level2">
<h2>Data analysis</h2>
<div id="statistical-hypothesis-test" class="section level3">
<h3>Statistical hypothesis test</h3>
<p>To test if there is a difference between the peptide abundances in
our two conditions (i.e. rapamycin treated and untreated) we use a
moderated t-test based on the <code>limma</code> R package.</p>
<p>Before the statistical hypothesis test we have to define the types of
missing values present in our data set. We are going to use the function
<code>assign_missingness()</code> which will return a column with
information on the types of missingness we have in our data
(i.e. complete, missing at random (MAR) or missing not at random
(MNAR)). We use the default parameters of this function which assumes
that missing values are MAR when the conditions are at least 70 %
complete (adjusted downward). Missing values are assumed to be MNAR when
less than 20 % of values are present (adjusted_downward) in one
condition if the other condition is complete. If not “complete” all
other comparisons are label as <code>NA</code>. If imputation is
performed, these are the comparisons that will not be imputed. The type
of missingness assigned to a comparison does not have any influence on
the statistical test. However, by default (can be changed) comparisons
with missingness <code>NA</code> are filtered out prior to p-value
adjustment. This means that in addition to imputation the user can use
missingness cutoffs also in order to define which comparisons are too
incomplete to be trustworthy even if significant.</p>
<p>After assigning the types of data missingness we use the function
<code>calculate_diff_abundance()</code>. By selecting
<code>method = t-test</code> the function will perform a Welch’s t-test.
There are also options included to perform a moderated t-test based on
the R package <code>limma</code> or to detect differential abundances
based on the algorithm implemented in the R package <code>proDA</code>.
The algorithm used for <code>proDA</code> is based on a probabilistic
dropout model which facilitates hypothesis testing (using a moderated
t-test) while eliminating the need for imputation.</p>
<p>It has been shown that generally moderated t-tests perform much
better also in proteomics data, as compared to t-tests (<a href="https://www.sciencedirect.com/science/article/pii/S2212968515000069">Kammers
et al. 2015</a>). Therefore, we will use a moderated t-test in this
example.</p>
<p>Please note that in this example we are not imputing missing values.
You can, however, do so by using the function <code>impute()</code>.
This function uses the output of <code>assign_missingness()</code> as
its input. You can use two different imputation methods:</p>
<ul>
<li><code>method = ludovic</code> will sample values that are MNAR from
a normal distribution around a value that is 3 (log2) lower than the
mean intensity of the non-missing condition. The method is was developed
by our colleague <a href="https://imsb.ethz.ch/research/picotti/PeoplePicotti/ludovicgillet.html">Ludovic
Gillet</a>.</li>
<li><code>method = noise</code> will sample MNAR values from a normal
distribution around the mean noise of the complete condition. This
requires you to have an additional column with information on the noise,
which can be obtained from Spectronaut.</li>
</ul>
<p>Both methods impute MAR data using the mean and variance of the
condition with the missing data. Missingness assigned as <code>NA</code>
will not be imputed.</p>
<p><em>Note: If data is imputed this can lead to invalid inferential
conclusions due to underestimating statistical uncertainty or it can
cause loss of statistical power (<a href="https://www.biorxiv.org/content/10.1101/661496v1.full.pdf">Ahlmann-Eltze
et al. 2020</a>). Therefore, we do not recommend using a moderated
t-test or the proDA algorithm after imputation.</em></p>
<p>Since we are dealing with a LiP-MS data set we perform the
statistical analysis on the precursor* level. For protein abundance data
you can simply use protein abundances as your intensities and select
your protein groups column for the grouping argument. Make sure to
retain any columns you need for further data analysis with the
<code>retain_columns</code> argument of both functions.</p>
<p>_*A peptide precursor is the actual molecular unit that was detected
on the mass spectrometer. This is a peptide with one specific charge
state and its modification(s)._</p>
<p><em>Note: Although it is not required for the data set analysed in
this vignette, analysis of LiP-MS data frequently requires correction of
LiP peptide intensities for changes in protein abundance. This can be
done using the steps outlined in <a href="https://doi.org/10.1038/nprot.2017.100">Schopper et
al. 2017</a>.</em></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>diff_abundance_data <span class="ot">&lt;-</span> data_filtered_uniprot <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign_missingness</span>(</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">sample =</span> r_file_name,</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">condition =</span> r_condition,</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">grouping =</span> eg_precursor_id,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">intensity =</span> normalised_intensity_log2,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">ref_condition =</span> <span class="st">&quot;control&quot;</span>,</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">completeness_MAR =</span> <span class="fl">0.7</span>,</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">completeness_MNAR =</span> <span class="fl">0.25</span>,</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_columns =</span> <span class="fu">c</span>(pg_protein_accessions, </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>                       go_f, </span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>                       xref_string, </span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>                       start, </span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>                       end, </span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>                       length, </span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>                       coverage)</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">calculate_diff_abundance</span>(</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">sample =</span> r_file_name,</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">condition =</span> r_condition,</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">grouping =</span> eg_precursor_id,</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">intensity_log2 =</span> normalised_intensity_log2,</span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">missingness =</span> missingness,</span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> comparison,</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;moderated_t-test&quot;</span>,</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_columns =</span> <span class="fu">c</span>(pg_protein_accessions, </span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>                       go_f, </span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>                       xref_string, </span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>                       start, </span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>                       end, </span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>                       length, </span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a>                       coverage)</span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a>  ) </span></code></pre></div>
</div>
<div id="p-value-distribution" class="section level3">
<h3>p-value distribution</h3>
<p>The p-value calculated with the moderated t-test is automatically
adjusted for multiple testing using the Benjamini-Hochberg correction.
This assures that we keep the false discovery rate low. An assumption of
this correction is however, that p-values should have an overall uniform
distribution. If there is an effect in the data, there will be an
increased frequency of low p-values. You can check this by using the
protti function <code>pval_distribution_plot()</code>. This also helps
you assess whether your p-value distribution fulfills the assumptions
for your selected FDR control. The <a href="https://CRAN.R-project.org/package=cp4p">cp4p</a> R package is
another great way to check the assumptions underlying FDR control in
quantitative experiments.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pval_distribution_plot</span>(<span class="at">data =</span> diff_abundance_data,</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">grouping =</span> eg_precursor_id,</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">pval =</span> pval</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>                       )</span></code></pre></div>
<p>For this subset of data the distribution of p-values is relatively
flat and there is no large increase in values in the low p-value range
(the distribution is uniform when a lot of your hypotheses are null).
This is likely because, for this experiment, only a very small fraction
of peptides show changes.</p>
<p>It is recommended to always check the (non-adjusted) p-value
distribution. A histogram that does not have a uniform distribution with
or without a specific enrichment for very low p-values (due to a
treatment induced effect) indicates a failure of the theoretical null
distribution, which could have several causes (<a href="https://www.cambridge.org/core/services/aop-cambridge-core/content/view/0BD692B5CFF1C51B73D480B72B0B3A28/9780511761362c6_p89-112_CBO.pdf/theoretical_permutation_and_empirical_null_distributions.pdf">Efron
2010, chapter 6</a>).</p>
<p>You can read more about p-value distributions <a href="http://varianceexplained.org/statistics/interpreting-pvalue-histogram/">here</a>.</p>
</div>
<div id="volcano-plot" class="section level3">
<h3>Volcano plot</h3>
<p>Next we are going to visualise the output of the previously performed
hypothesis test to assess the results of our experiment. For this we are
going to plot a volcano plot with fold-changes on the x-axis and the
p-value on the y-axis. The output of the previously used
<code>calculate_diff_abundance()</code> function is ideal to use for the
<code>volcano_plot()</code> function as it contains all the information
we need: precursor IDs, protein IDs, fold changes (<code>diff</code>),
p-values (<code>pval</code>) and adjusted p-values
(<code>adj_pval</code>). We are going to highlight the peptides of the
known target of rapamycin FKBP12 (UniProt ID = P62942) in blue to
quickly find the peptides in the plot. You can also make the plot
interactive by setting <code>interactive = TRUE</code>. This will help
you quickly obtain more information on each point in the plot.</p>
<p>Since adjusted p-values are related to unadjusted p-values often in a
complex way, it makes them hard to be interpret if they would be used
for the y-axis. To nevertheless use the information of adjusted p-values
for the plot, you can provide the column name of the adjusted p-values
to the <code>significance_cutoff</code> argument next to the desired
cutoff. The function will look for the closest adjusted p-values above
and below the set cutoff and take the mean of the corresponding p-value
as the cutoff line. If there is no adjusted p-value in the data that is
below the set cutoff no line is displayed. This allows you to display
volcano plots using p-values while using adjusted p-values for the
cutoff criteria.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">volcano_plot</span>(</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> diff_abundance_data,</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">grouping =</span> eg_precursor_id,</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">log2FC =</span> diff,</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">significance =</span> pval,</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;target&quot;</span>,</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">target_column =</span> pg_protein_accessions,</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">target =</span> <span class="st">&quot;P62942&quot;</span>,</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_axis_label =</span> <span class="st">&quot;log2(fold change) Rapamycin treated vs. untreated&quot;</span>,</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">significance_cutoff =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="st">&quot;adj_pval&quot;</span>) </span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="co"># The significance_cutoff argument can also just be used for a </span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="co"># regular cutoff line by just providing the cutoff value, e.g.</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="co"># signficiance_cutoff = 0.05</span></span></code></pre></div>
</div>
<div id="barcode-plot" class="section level3">
<h3>Barcode plot</h3>
<p>For LiP-MS experiments a good way to see where on the protein the
changes due to binding or conformational changes occur is to plot a
barcode plot. A barcode plot can be created with the
<strong>protti</strong> function <code>barcode_plot()</code>. The
detected peptides are coloured in grey and the changing peptides are
highlighted in blue.</p>
<p>In order to produce a barcode plot only for our target FKBP12 we
create a data frame that contains only information for our target
protein using <code>dplyr</code>’s <code>filter()</code> function. The
filtered data frame is then used as the input for the plot.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>FKBP12 <span class="ot">&lt;-</span> diff_abundance_data <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pg_protein_accessions <span class="sc">==</span> <span class="st">&quot;P62942&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">barcode_plot</span>(</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> FKBP12,</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">start_position =</span> start,</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">end_position =</span> end,</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">protein_length =</span> length,</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">coverage =</span> coverage,</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">colouring =</span> diff,</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">cutoffs =</span> <span class="fu">c</span>(<span class="at">diff =</span> <span class="dv">1</span>, <span class="at">adj_pval =</span> <span class="fl">0.05</span>),</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">protein_id =</span> pg_protein_accessions</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
</div>
<div id="woods-plot" class="section level3">
<h3>Wood’s plot</h3>
<p>An additional way to plot LiP-MS changes is the Woods’ plot. This
plot will show the extent of the precursor fold changes along the
protein sequence. The precursors are located on the x-axis based on
their start and end positions. The position on the y-axis displays the
fold change. The vertical size (y-axis) of the box representing the
precurors does not have any meaning.</p>
<p>To produce a Woods’ plot we use the function
<code>woods_plot()</code> and colour the peptides according to their
adjusted p-values. We are highlighting significant adjusted p-values
(&lt; 0.01) with an asterisk. Peptides can also be coloured by another
categorical or continous variable. Asterisks can be added for any
logical (binary) variable.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>FKBP12 <span class="ot">&lt;-</span> FKBP12 <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">significant =</span> <span class="fu">ifelse</span>(adj_pval <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>))</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">woods_plot</span>(</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> FKBP12,</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">fold_change =</span> diff,</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">start_position =</span> start,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">end_position =</span> end,</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">protein_length =</span> length,</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">coverage =</span> coverage,</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">colouring =</span> adj_pval,</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">protein_id =</span> pg_protein_accessions, </span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">facet =</span> <span class="cn">FALSE</span>,</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">fold_change_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">highlight =</span> significant</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
</div>
<div id="peptide-profile-plots" class="section level3">
<h3>Peptide profile plots</h3>
<p>To see how the individual precursors in our target protein are
changing with the treatment we plot profile plots by using the function
<code>peptide_profile_plot()</code>. This is particularly useful as you
can quickly see if your whole protein changes in abundance or only a
fraction of precursors/peptides. If you have protein abundance data you
can also use the plot to show changes in protein abundance over your
treatment condition(s). By selecting multiple targets (as a vector) you
can produce the plot for multiple proteins.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>FKBP12_intensity <span class="ot">&lt;-</span> data_filtered_uniprot <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pg_protein_accessions <span class="sc">==</span> <span class="st">&quot;P62942&quot;</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fu">peptide_profile_plot</span>(</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> FKBP12_intensity,</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample =</span> r_file_name,</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">peptide =</span> eg_precursor_id,</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">intensity_log2 =</span> normalised_intensity_log2,</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">grouping =</span> pg_protein_accessions,</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">targets =</span> <span class="st">&quot;P62942&quot;</span>,</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">protein_abundance_plot =</span> <span class="cn">FALSE</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
</div>
<div id="additional-helpful-functions" class="section level2">
<h2>Additional helpful functions</h2>
<p><strong>protti</strong> includes additional helpful functions that do
not make sense to use for this data set but apply to data sets of full
size that have global changes. These functions include the
<code>calculate_go_enrichment()</code>function that helps you check if
your hits are enriched for specific gene ontology (GO) terms, the
<code>analyse_functional_network()</code> function that plots a String
network based on information from StringDB for your hits and the
<code>calculate_kegg_enrichment()</code> function which checks for
enriched pathways in your hits. Furthermore, you can directly check for
enrichment of a self defined treatment with the
<code>calculate_treatment_enrichment()</code> function.</p>
<p>For GO enrichment you would add an additional column to your data
frame containing information on whether your hit is significant or not.
You can do this by using the <code>dplyr</code> function
<code>mutate()</code>. Here we want the column to contain logicals that
are either TRUE when the adjusted p-value is below 0.05 and the
log2(fold change) is below -1 or above 1 or to be FALSE if this is not
the case. We use the <code>ifelse()</code> function to produce the
logicals. Furthermore, we annotate if the hit is true positive by
marking peptides of the known rapamycin binding protein FKBP12.</p>
<p>For the network analysis we filter the previously produced data frame
containing the <code>is_significant</code> column for significant hits.
This data frame can then be used as an input for
<code>analyse_functional_network()</code> to check if the proteins can
be found in an interaction network based on StringDB information.</p>
<p>For <code>calculate_kegg_enrichment()</code> you need to first use
the function <code>fetch_kegg()</code> to obtain the KEGG pathway
identifiers for your data set. You can then use <code>dplyr</code>’s
<code>right_join()</code>to join the output with the previously produced
data frame containing a column indicating whether your hits are
significant or not.</p>
<p>If you know all known interactors of your specific treatment you can
check for an enrichment of these with the
<code>calculate_treatment_enrichment()</code> function. This is
particularly useful if your treatment has an effect on many
proteins.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>diff_abundance_significant <span class="ot">&lt;-</span> diff_abundance_data <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mark significant peptides</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">is_significant =</span> <span class="fu">ifelse</span>((adj_pval <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(diff) <span class="sc">&gt;</span> <span class="dv">1</span>), <span class="cn">TRUE</span>, <span class="cn">FALSE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mark true positive hits</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">binds_treatment =</span> pg_protein_accessions <span class="sc">==</span> <span class="st">&quot;P62942&quot;</span>) </span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="do">### GO enrichment using &quot;molecular function&quot; annotation from UniProt</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="fu">calculate_go_enrichment</span>(</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> diff_abundance_significant,</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">protein_id =</span> pg_protein_accessions,</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">is_significant =</span> is_significant,</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">go_annotations_uniprot =</span> go_f</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="do">### Network analysis</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>network_input <span class="ot">&lt;-</span> diff_abundance_significant <span class="sc">%&gt;%</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(is_significant <span class="sc">==</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a><span class="fu">analyse_functional_network</span>(<span class="at">data =</span> network_input,</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>                 <span class="at">protein_id =</span> pg_protein_accessions,</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>                 <span class="at">string_id =</span> xref_string,</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>                 <span class="at">binds_treatment =</span> binds_treatment,</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>                 <span class="at">organism_id =</span> <span class="dv">9606</span>)</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a><span class="do">### KEGG pathway enrichment</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a><span class="co"># First you need to load KEGG pathway annotations from the KEGG database </span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a><span class="co"># for your specific organism of interest. In this case HeLa cells were </span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a><span class="co"># used, therefore the organism of interest is homo sapiens (hsa)</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">fetch_kegg</span>(<span class="at">species =</span> <span class="st">&quot;hsa&quot;</span>)</span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Next we need to annotate our data with KEGG pathway IDs and perform enrichment analysis</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>diff_abundance_significant <span class="sc">%&gt;%</span> </span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># columns containing proteins IDs are named differently</span></span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(kegg, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;pg_protein_accessions&quot;</span> <span class="ot">=</span> <span class="st">&quot;uniprot_id&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">calculate_kegg_enrichment</span>(<span class="at">protein_id =</span> pg_protein_accessions,</span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>                  <span class="at">is_significant =</span> is_significant,</span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>                  <span class="at">pathway_id =</span> pathway_id,</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>                  <span class="at">pathway_name =</span> pathway_name)</span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a><span class="do">### Treatment enrichment analysis</span></span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a><span class="fu">calculate_treatment_enrichment</span>(diff_abundance_significant,</span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>                     <span class="at">protein_id =</span> pg_protein_accessions,</span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>                     <span class="at">is_significant =</span> is_significant,</span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>                     <span class="at">binds_treatment =</span> binds_treatment,</span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a>                     <span class="at">treatment_name =</span> <span class="st">&quot;Rapamycin&quot;</span>)</span></code></pre></div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
