<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Jan-Philipp Quast" />

<meta name="date" content="2024-07-12" />

<title>Dose-Response Data Analysis Workflow</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Dose-Response Data Analysis Workflow</h1>
<h4 class="author">Jan-Philipp Quast</h4>
<h4 class="date">2024-07-12</h4>



<style>
body {
text-align: justify}
</style>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<pre><code>#&gt; [1] &quot;!!! IMPORTANT !!!&quot;
#&gt; [1] &quot;This Vignette has not been built completely on CRAN due to size limitations.&quot;
#&gt; [1] &quot;Please check the correct version here: &quot;
#&gt; [1] &quot;https://jpquast.github.io/protti/articles/https://jpquast.github.io/protti/articles/data_analysis_dose_response_workflow.html&quot;</code></pre>
<p>This vignette will take you through all important steps for the
analysis of dose-response experiments with <strong>protti</strong>. For
experiments with treatment titrations (i.e. samples treated with
different amounts of the same compound, protein, metabolite, element,
ion, etc.) you can fit sigmoidal log-logistic regression curves to the
data and determine significantly changing precursors, peptides or
proteins based on Pearson correlation and ANOVA q-value.</p>
<p>Before analysing your data please ensure that it is of sufficient
quality by using <strong>protti</strong>’s quality control functions
(see <a href="https://jpquast.github.io/protti/articles/quality_control_workflow.html">Quality
control (QC) workflow</a>) If you would like to compare a control
condition with one treatment condition (or multiple unrelated
treatments) you can check out the <a href="https://jpquast.github.io/protti/articles/data_analysis_single_dose_treatment_workflow.html">data
analysis vignette for single dose treatment data</a>. For help with data
input please click <a href="https://jpquast.github.io/protti/articles/input_preparation_workflow.html">here</a>.</p>
<p>A typical dose-response experiment contains multiple samples that
were treated with different amounts of e.g. a drug. Replicates of
samples treated with the same dose make up a condition. Commonly, the
first concentration is 0 (i.e. the control, in which treatment is with
the solvent of the drug). Dose-response treatments require a minimal
number of treatments to fit curves with sufficient quality. For analysis
with <strong>protti</strong> at least 5 different conditions should be
present. Another consideration for your experiment is the range of
concentrations. They should not be too close together since effects are
usually best identified over a larger concentration range. But you
should make sure not to space them out too far. It is generally advised
to space them evenly on a logarithmic scale of the base 10 or <span class="math inline">\(e\)</span> (Euler’s number). You can also include
steps in between e.g. 100, 500, 1000 or 100, 200, 1000 for a log10
scale. It is advisable to use a rather broad concentration range for an
experiment in which you do not know what to expect.</p>
<p><strong>protti</strong> fits four-parameter log-logistic
dose-response models to your data. It utilizes the <code>drm()</code>
and <code>LL.4()</code> functions from the <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0146021"><code>drc</code></a>
package for this. You can also select a non logarithmic model using the
<code>L.4()</code> function in case your data does not follow a
log-logistic but a logistic regression.</p>
<p>For limited proteolysis-coupled to mass spectrometry (<a href="https://www.nature.com/articles/nbt.2999">LiP-MS</a>) data,
dose-response curves have been used previously for the identification of
drug binding sites in complex proteomes (<a href="https://www.nature.com/articles/s41467-020-18071-x">Piazza
2020</a>). Since LiP-MS data is analysed on the peptide or precursor*
level, using additional information about peptide behaviour from
multiple conditions reduces false discovery rate.</p>
<p><em>A peptide precursor is the actual molecular unit that was
detected on the mass spectrometer. This is a peptide with one specific
charge state and its modification(s).</em></p>
</div>
<div id="getting-started" class="section level1">
<h1>Getting started</h1>
<p>Before starting the analysis of your data you need to load
<strong>protti</strong> and additional packages used for the analysis.
As described on the <a href="https://jpquast.github.io/protti/index.html">main page</a> of
<strong>protti</strong>, the <a href="https://www.tidyverse.org">tidyverse</a> package collection works
well together with the rest of the package, by design. You can load
packages after you installed them with the <code>library()</code>
function.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># Load packages</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(protti)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code></pre></div>
<div id="loading-data" class="section level2">
<h2>Loading data</h2>
<p>For this vignette we use a subset of proteins from an experiment of
HeLa cell lysates treated with 9 doses of rapamycin followed by LiP-MS.
Rapamycin forms a complex with the FK506-binding protein (FKBP12) that
binds and allosterically inhibits mTORC1 (<a href="https://www.sciencedirect.com/science/article/pii/0092867494905703?via%3Dihub">Sabatini
1994</a>). Since rapamycin is known to be a highly specific drug, we
expect to identify FKBP12 as one of the only interacting proteins.</p>
<p>We included 39 random proteins and FKBP12 in this sample data set.
The proteins were sampled using the seed 123.</p>
<p>If you want to read your data into R, we suggest using the
<code>read_protti()</code> function. This function is a wrapper around
the fast <code>fread()</code> function from the <a href="https://rdatatable.gitlab.io/data.table/"><code>data.table</code></a>
package and the <code>clean_names()</code> function from the <a href="https://sfirke.github.io/janitor/"><code>janitor</code></a>
package. This will allow you to not only load your data into R very
fast, but also to clean up the column names into lower snake case. This
will make it easier to remember them and to use them in your data
analysis. To ensure that your data input has the right format you can
check out our <a href="https://jpquast.github.io/protti/articles/input_preparation_workflow.html">input
preparation vignette</a>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>rapamycin_dose_response <span class="ot">&lt;-</span> <span class="fu">read_protti</span>(<span class="st">&quot;your_data.csv&quot;</span>)</span></code></pre></div>
<p>In this example the rapamycin dose-response data set is included in
protti and you can easily use it by calling the <code>data()</code>
function. This will load the data into your R session.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>utils<span class="sc">::</span><span class="fu">data</span>(<span class="st">&quot;rapamycin_dose_response&quot;</span>)</span></code></pre></div>
</div>
<div id="cleaning-data" class="section level2">
<h2>Cleaning data</h2>
<p>After loading data into R we would like to clean it up to remove rows
that are problematic for our analysis. The
<code>rapamycin_dose_response</code> data is based on a report from <a href="https://biognosys.com/software/spectronaut/">Spectronaut™</a>. It
includes convenient columns that label peptides as being decoys
(<code>eg_is_decoy</code>). Other software reports might have similar
columns that should be checked. MaxQuant for example includes
information on contaminating proteins that you should remove prior to
data analysis. In this case we would like to remove any peptides left in
the report that are decoys (used for false discovery rate estimation).
It is a good practice to just run this even though decoys are usually
already filtered out.</p>
<p>Due to the fact that with increasing raw intensities
(<code>fg_quantity</code>) also variances increase, statistical tests
would have a bias towards lower intense peptides or proteins. Therefore
you should log2 transform your data to correct for this mean-variance
relationship. We do this by using <code>dplyr</code>’s
<code>mutate()</code> together with <code>log2()</code>.</p>
<p>In addition to filtering and log2 transformation it is also advised
to normalise your data to equal out small differences in overall sample
intensities that result from unequal sample concentrations.
<strong>protti</strong> provides the <code>normalise()</code> function
for this purpose. For this example we will use median normalisation
(<code>method = &quot;median&quot;</code>). This function generates an additional
column called <code>normalised_intensity_log2</code> that contains the
normalised intensities.</p>
<p><em>Note: If your search tool already normalised your data you should
not normalise it again.</em></p>
<p>In addition to the removal of decoys we also remove any
non-proteotypic peptides (<code>pep_is_proteotypic</code>) from our
analysis. This is specific for the analysis of LiP-MS data. If a peptide
is non-proteotypic it is part of two or more proteins. If we detect a
change in non-proteotypic peptides it is not possible to clearly assign
which of the proteins it is coming from.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co"># Filter, log2 transform and normalise data</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>data_normalised <span class="ot">&lt;-</span> rapamycin_dose_response <span class="sc">%&gt;%</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="fu">filter</span>(eg_is_decoy <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">intensity_log2 =</span> <span class="fu">log2</span>(fg_quantity)) <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  <span class="fu">normalise</span>(</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>    <span class="at">sample =</span> r_file_name,</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>    <span class="at">intensity_log2 =</span> intensity_log2,</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;median&quot;</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>  <span class="fu">filter</span>(pep_is_proteotypic <span class="sc">==</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>It is also useful to check the intensity distribution of all
precursors with the <strong>protti</strong> function
<code>qc_intensity_distribution()</code>. Usually, the distribution of
intensities is a skewed normal distribution of values missing on the
left flank. This is due to the detection limit of the mass spectrometer.
The probability of not observing a value gets higher with decreasing
intensities.</p>
<p>For experiments measured in data independent acquisition (DIA) mode,
it is good to filter out very low intensity values that are not part of
the distribution. These values are likely false assignments of peaks. If
you look closely you can see a small peak around 0 corresponding to
these values. In the case of our distribution we could chose a cutoff at
a log2 intensity of 5, which corresponds to a raw intensity of 32.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">qc_intensity_distribution</span>(</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  <span class="at">data =</span> data_normalised,</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  <span class="at">grouping =</span> eg_precursor_id,</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>  <span class="at">intensity =</span> normalised_intensity_log2,</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>  <span class="at">plot_style =</span> <span class="st">&quot;histogram&quot;</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>data_normalised <span class="ot">&lt;-</span> data_normalised <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="fu">filter</span>(normalised_intensity_log2 <span class="sc">&gt;</span> <span class="dv">5</span>)</span></code></pre></div>
</div>
<div id="clustering-of-samples" class="section level2">
<h2>Clustering of samples</h2>
<p>Before we fit dose-response models to our data, we can check how
samples cluster. Ideally, samples that belong to the same treatment
concentration cluster together. A common way that you can analyse this
is correlation based hierachical clustering using the
<code>qc_sample_correlation()</code> function from
<strong>protti</strong>. The correlation of samples is calculated based
on the correlation of precursor (or peptide/protein) intensities. By
default <code>qc_sample_correlation()</code> uses Spearman correlation,
however you can change that to any correlation supported by the
<code>cor()</code> function.</p>
<p><em>Note: Many of <strong>protti</strong>’s plotting functions also
have the option to display an interactive version of the plot with an
<code>interactive</code> argument.</em></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">qc_sample_correlation</span>(</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  <span class="at">data =</span> data_normalised,</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  <span class="at">sample =</span> r_file_name,</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>  <span class="at">grouping =</span> eg_precursor_id,</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>  <span class="at">intensity_log2 =</span> normalised_intensity_log2,</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>  <span class="at">condition =</span> r_condition</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>)</span></code></pre></div>
<p>For our rapamycin data set we cannot identify any clear sample
clustering based on correlation, because only a small subset of proteins
was selected for this example. Furthermore, data sets in which only a
few changes are expected commonly do not cluster nicely. This is because
precursors/peptides/proteins have very similar intensity values across
conditions and clustering is in this case based on variance. In case of
more global changes samples should usually cluster nicely.</p>
<p>In addition to correlation based clustering a principal component
analysis can be performed using the <strong>protti</strong> function
<code>qc_pca()</code>. Similar to before, good clustering is usually
dependent on the amount of changing precursors (or
peptides/proteins).</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">qc_pca</span>(</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="at">data =</span> data_normalised,</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="at">sample =</span> r_file_name,</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="at">grouping =</span> eg_precursor_id,</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>  <span class="at">intensity =</span> normalised_intensity_log2,</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>  <span class="at">condition =</span> r_condition</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>)</span></code></pre></div>
</div>
</div>
<div id="fitting-dose-response-curves" class="section level1">
<h1>Fitting dose-response curves</h1>
<p>The main part of a dose-response data analysis is the
<code>fit_drc_4p()</code> function. It fits four-parameter log-logistic
dose-response curves for every precursor (or peptide/protein) with the
following equation:</p>
<p><span class="math display">\[y = d + \frac{a - d}{1 +
(\frac{x}{c})^{-b}}\]</span> <span class="math inline">\(x\)</span> is
the dependent variable<br />
<span class="math inline">\(y\)</span> is the independen variable<br />
<span class="math inline">\(a\)</span> is the minimum value that can be
obtained<br />
<span class="math inline">\(d\)</span> is the maximum value that can be
obtained<br />
<span class="math inline">\(c\)</span> is the point of inflection<br />
<span class="math inline">\(b\)</span> is the Hill’s coefficient
(e.g. the negative slope at the inflection point)</p>
<p>The output of <code>fit_drc_4p()</code> provides extensive
information on the goodness of fit. Furthermore, the function filters
and ranks fits based on several criteria which include completeness of
data and a significance cutoff based on an adjusted p-value obtained
from ANOVA. For details about the filtering steps you can read the
function documentation by calling <code>?fit_drc_4p</code>. If you only
care about your potential hits (and exclude precursors/peptides/proteins
that do not pass the filtering e.g. due to too few observations), you
can choose <code>filter = &quot;pre&quot;</code>, which filters the data before
model fitting. This speeds up the process because less models need to be
fit. If you want to perform enrichment analysis later on you should keep
the default <code>filter = &quot;post&quot;</code>. This will fit all possible
models and then only annotate models based on whether they passed or
failed the filtering step (<code>passed_filter</code>).</p>
<p>Since <strong>protti</strong> version <code>0.8.0</code> we recommend
you use the <code>n_replicate_completeness</code> and
<code>n_condition_completeness</code> arguments in order to define
cutoffs for completeness of replicates in a given number of conditions.
This makes the output of the function reproducible independent of the
input data. With the provided cutoffs below we define that at least 2
replicates need to be detected in 4 conditions. Any precursor with less
completeness than that will not be considered and removed from the
data.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> data_normalised <span class="sc">%&gt;%</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="fu">fit_drc_4p</span>(</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>    <span class="at">sample =</span> r_file_name,</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>    <span class="at">grouping =</span> eg_precursor_id,</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>    <span class="at">response =</span> normalised_intensity_log2,</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>    <span class="at">dose =</span> r_condition,</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>    <span class="at">filter =</span> <span class="st">&quot;post&quot;</span>,</span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>    <span class="at">n_replicate_completeness =</span> <span class="dv">2</span>, <span class="co"># specifiy this based on your data and cutoff criteria</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>    <span class="at">n_condition_completeness =</span> <span class="dv">4</span>, <span class="co"># specifiy this based on your data and cutoff criteria</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>    <span class="at">retain_columns =</span> <span class="fu">c</span>(pg_protein_accessions)</span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>  )</span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a><span class="co"># make sure to retain columns that you need later but that are not part of the function</span></span></code></pre></div>
<p>If you have to fit many models this function may take about an hour
or longer. If you have the computational resources you can speed this up
by using <code>parallel_fit_drc_4p()</code>, a parallelised version of
this function.</p>
<p><em>Note: Keep in mind that by spreading the model fitting over
multiple cores, more memory than usual is needed. Furthermore, using the
maximal number of available cores may not be the most efficient, because
it takes time to copy the data to and from each core - this process will
not be performed in parallel. Therefore, it is advised that the run time
on each core is at least as long as the time to set up (data transfer,
environment preparation etc.) all cores.</em></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="co"># setup of cores. Make sure you have the future package installed</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>future<span class="sc">::</span><span class="fu">plan</span>(future<span class="sc">::</span>multisession, <span class="at">workers =</span> <span class="dv">3</span>)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co"># fit models in parallel</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>parallel_fit <span class="ot">&lt;-</span> data_normalised <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>  <span class="fu">parallel_fit_drc_4p</span>(</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>    <span class="at">sample =</span> r_file_name,</span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>    <span class="at">grouping =</span> eg_precursor_id,</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a>    <span class="at">response =</span> normalised_intensity_log2,</span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a>    <span class="at">dose =</span> r_condition,</span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a>    <span class="at">retain_columns =</span> <span class="fu">c</span>(pg_protein_accessions),</span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a>    <span class="at">n_cores =</span> <span class="dv">3</span>,</span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a>    <span class="at">n_replicate_completeness =</span> <span class="dv">2</span>, <span class="co"># specifiy this based on your data and cutoff criteria</span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a>    <span class="at">n_condition_completeness =</span> <span class="dv">4</span>, <span class="co"># specifiy this based on your data and cutoff criteria</span></span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a>  )</span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a></span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a><span class="co"># remove workers again after you are done</span></span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a>future<span class="sc">::</span><span class="fu">plan</span>(future<span class="sc">::</span>sequential)</span></code></pre></div>
<p>If we examine all precursors based on their rank (calculated from
correlation and ANOVA q-value), we can see that as expected most of them
are FKBP12 (P62942) peptides. One benefit of analysing LiP-MS data on
the precursor level (rather than peptide level) is that multiple lines
of evidence for a change in the specific peptide can be used (we have
data for each different charge and modification state). Therefore, it is
always good to check if there are other precursors of a good scoring
peptide that do not show any regulation at all. This would mean that the
reason for the observed response is not based on a biological
effect.</p>
<div id="model-fit-plotting" class="section level2">
<h2>Model fit plotting</h2>
<p>After model fitting, you can easily visualise model fits by plotting
them with the <code>drc_4p_plot()</code> function. You can provide one
or more precursor IDs to the <code>targets</code> argument or just
return plots for all precursors with <code>targets = &quot;all&quot;</code>. Make
sure that the number of plots is reasonable to return within R. You can
also directly export your plots by setting
<code>export = TRUE</code>.</p>
<p><em>Note: The output of <code>fit_drc_4p()</code> includes two
columns with data frames containing the model and original data points
required for the plot. These data frames contain the original names for
<code>dose</code> and <code>response</code> that you provided to
<code>fit_drc_4p()</code>.</em></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="co"># Model plotting</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="fu">drc_4p_plot</span>(fit,</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>  <span class="at">grouping =</span> eg_precursor_id,</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>  <span class="at">dose =</span> r_condition,</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>  <span class="at">response =</span> normalised_intensity_log2,</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>  <span class="at">targets =</span> <span class="st">&quot;_VFDVELLKLE_.2&quot;</span>,</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>  <span class="at">unit =</span> <span class="st">&quot;pM&quot;</span>,</span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>  <span class="at">x_axis_limits =</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="fl">1e+08</span>),</span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>  <span class="at">export =</span> <span class="cn">FALSE</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>)</span></code></pre></div>
<p>There are additional arguments that help you customise your plot. You
can for example specify the axis limits with <code>x_axis_limits</code>.
This should generally always be set to the lowest non-zero and highest
concentration <code>x_axis_limits = c(10, 1e+08)</code>. Like this you
can more easily see if extreme concentrations are completely missing
from the curve. The <code>colours</code> argument allows you to specify
your own colours for the points, confidence interval and curve.</p>
</div>
</div>
<div id="further-analysis" class="section level1">
<h1>Further analysis</h1>
<p>After models are fit, you can have a deeper look into the hits you
obtained. <strong>protti</strong> provides several functions that help
you to identify patterns in your data.</p>
<p>You can test your data set for gene ontology (GO) term enrichment
(<code>calculate_go_enrichment()</code>). Alternatively, pathway
annotations provided in the KEGG database can be used to test for
pathway enrichments (<code>calculate_kegg_enrichment()</code>).</p>
<p>If you know which proteins bind or interact with your specific
treatment, you can provide your own list of true positive hits and check
if these are enriched in your significant hits by using
<strong>protti</strong>’s <code>calculate_treatment_enrichment()</code>
function. For our LiP-MS experiment using rapamycin we are probing
direct interaction with proteins in contrast to functional effects. The
only protein that rapamycin is known to bind to is FKBP12 and testing
the significance of enrichment for a single or even a few proteins is
not appropriate. However, testing for enrichment is especially useful if
your treatment affects many proteins, since it can help you to reduce
the complexity of your result.</p>
<p>The <a href="https://academic.oup.com/nar/article/47/D1/D607/5198476">STRING</a>
database provides a good resource for the analysis of protein
interaction networks. It is often very useful to check for interactions
within your significant hits. For LiP-MS data this sometimes explains
why proteins that do not directly interact with your treatment are still
significantly affected. With <code>analyse_functional_network()</code>,
<strong>protti</strong> provides a useful wrapper around some <a href="https://www.bioconductor.org/packages/release/bioc/html/STRINGdb.html"><code>STRINGdb</code></a>
package functions.</p>
<div id="annotation-of-data" class="section level2">
<h2>Annotation of data</h2>
<p><strong>protti</strong> has several functions for fetching database
information. One commonly used database in the proteomics community is
UniProt. By providing the <code>fetch_uniprot()</code> function with a
character vector of your protein IDs, you can fetch additional
information about the proteins. These include names, IDs from other
databases, gene ontology annotations, information about the active
metabolite and metal binding sites, protein length and sequence. You can
freely select which information you are most interested in by providing
it to the <code>columns</code> arguments of the
<code>fetch_uniprot()</code> function based on the information they <a href="https://www.uniprot.org/help/return_fields">provide</a> for
download.</p>
<p>Furthermore, we will add a <code>passed_filter</code> column to the
data that contains TRUE or FALSE information based on if a precursor
passed the filtering criterion defined in the <code>fit_drc_4p()</code>
function. This is the case if the peptide was assigned a rank.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="co"># fetching of UniProt information</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>unis <span class="ot">&lt;-</span> <span class="fu">unique</span>(fit<span class="sc">$</span>pg_protein_accessions)</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>uniprot <span class="ot">&lt;-</span> <span class="fu">fetch_uniprot</span>(unis)</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a><span class="co"># annotation of fit data based on information from UniProt</span></span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>fit_annotated <span class="ot">&lt;-</span> fit <span class="sc">%&gt;%</span></span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>  <span class="co"># columns containing proteins IDs are named differently</span></span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>  <span class="fu">left_join</span>(uniprot, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;pg_protein_accessions&quot;</span> <span class="ot">=</span> <span class="st">&quot;accession&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a>  <span class="co"># mark peptides that pass the filtering</span></span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">passed_filter =</span> <span class="sc">!</span><span class="fu">is.na</span>(rank)) <span class="sc">%&gt;%</span></span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a>  <span class="co"># create new column with prior knowledge about binding partners of treatment</span></span>
<span id="cb13-12"><a href="#cb13-12" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">binds_treatment =</span> pg_protein_accessions <span class="sc">==</span> <span class="st">&quot;P62942&quot;</span>)</span></code></pre></div>
</div>
<div id="enrichment-and-network-analysis" class="section level2">
<h2>Enrichment and network analysis</h2>
<p>As mentioned earlier, for our test data set, enrichment or network
analysis are not useful, since we only use a very small random subset of
the complete data set and rapamycin only interacts with FKBP12.</p>
<p>Nevertheless, we will demonstrate below how you could use some
additional functions on your own data.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="do">### GO enrichment using &quot;molecular function&quot; annotation from UniProt</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="fu">calculate_go_enrichment</span>(fit_annotated,</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  <span class="at">protein_id =</span> pg_protein_accessions,</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>  <span class="at">is_significant =</span> passed_filter,</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>  <span class="at">go_annotations_uniprot =</span> go_f</span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>) <span class="co"># column obtained from UniProt</span></span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a><span class="do">### KEGG pathway enrichment</span></span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a><span class="co"># First you need to load KEGG pathway annotations from the KEGG database</span></span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a><span class="co"># for your specific organism of interest. In this case HeLa cells were</span></span>
<span id="cb14-13"><a href="#cb14-13" tabindex="-1"></a><span class="co"># used, therefore the organism of interest is homo sapiens (hsa)</span></span>
<span id="cb14-14"><a href="#cb14-14" tabindex="-1"></a></span>
<span id="cb14-15"><a href="#cb14-15" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">fetch_kegg</span>(<span class="at">species =</span> <span class="st">&quot;hsa&quot;</span>)</span>
<span id="cb14-16"><a href="#cb14-16" tabindex="-1"></a></span>
<span id="cb14-17"><a href="#cb14-17" tabindex="-1"></a><span class="co"># Next we need to annotate our data with KEGG pathway IDs and perform enrichment analysis</span></span>
<span id="cb14-18"><a href="#cb14-18" tabindex="-1"></a></span>
<span id="cb14-19"><a href="#cb14-19" tabindex="-1"></a>fit <span class="sc">%&gt;%</span></span>
<span id="cb14-20"><a href="#cb14-20" tabindex="-1"></a>  <span class="co"># columns containing proteins IDs are named differently</span></span>
<span id="cb14-21"><a href="#cb14-21" tabindex="-1"></a>  <span class="fu">left_join</span>(kegg, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;pg_protein_accessions&quot;</span> <span class="ot">=</span> <span class="st">&quot;uniprot_id&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-22"><a href="#cb14-22" tabindex="-1"></a>  <span class="fu">calculate_kegg_enrichment</span>(</span>
<span id="cb14-23"><a href="#cb14-23" tabindex="-1"></a>    <span class="at">protein_id =</span> pg_protein_accessions,</span>
<span id="cb14-24"><a href="#cb14-24" tabindex="-1"></a>    <span class="at">is_significant =</span> passed_filter,</span>
<span id="cb14-25"><a href="#cb14-25" tabindex="-1"></a>    <span class="at">pathway_id =</span> pathway_id, <span class="co"># column name from kegg data frame</span></span>
<span id="cb14-26"><a href="#cb14-26" tabindex="-1"></a>    <span class="at">pathway_name =</span> pathway_name</span>
<span id="cb14-27"><a href="#cb14-27" tabindex="-1"></a>  ) <span class="co"># column name from kegg data frame</span></span>
<span id="cb14-28"><a href="#cb14-28" tabindex="-1"></a></span>
<span id="cb14-29"><a href="#cb14-29" tabindex="-1"></a><span class="do">### Treatment enrichment analysis</span></span>
<span id="cb14-30"><a href="#cb14-30" tabindex="-1"></a></span>
<span id="cb14-31"><a href="#cb14-31" tabindex="-1"></a><span class="fu">calculate_treatment_enrichment</span>(fit_annotated,</span>
<span id="cb14-32"><a href="#cb14-32" tabindex="-1"></a>  <span class="at">protein_id =</span> pg_protein_accessions,</span>
<span id="cb14-33"><a href="#cb14-33" tabindex="-1"></a>  <span class="at">is_significant =</span> passed_filter,</span>
<span id="cb14-34"><a href="#cb14-34" tabindex="-1"></a>  <span class="at">binds_treatment =</span> binds_treatment,</span>
<span id="cb14-35"><a href="#cb14-35" tabindex="-1"></a>  <span class="at">treatment_name =</span> <span class="st">&quot;Rapamycin&quot;</span></span>
<span id="cb14-36"><a href="#cb14-36" tabindex="-1"></a>)</span>
<span id="cb14-37"><a href="#cb14-37" tabindex="-1"></a></span>
<span id="cb14-38"><a href="#cb14-38" tabindex="-1"></a><span class="do">### Network analysis</span></span>
<span id="cb14-39"><a href="#cb14-39" tabindex="-1"></a></span>
<span id="cb14-40"><a href="#cb14-40" tabindex="-1"></a>fit_annotated <span class="sc">%&gt;%</span></span>
<span id="cb14-41"><a href="#cb14-41" tabindex="-1"></a>  <span class="fu">filter</span>(passed_filter <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="co"># only analyse hits that were significant</span></span>
<span id="cb14-42"><a href="#cb14-42" tabindex="-1"></a>  <span class="fu">analyse_functional_network</span>(</span>
<span id="cb14-43"><a href="#cb14-43" tabindex="-1"></a>    <span class="at">protein_id =</span> pg_protein_accessions,</span>
<span id="cb14-44"><a href="#cb14-44" tabindex="-1"></a>    <span class="at">string_id =</span> xref_string, <span class="co"># column from UniProt containing STRING IDs</span></span>
<span id="cb14-45"><a href="#cb14-45" tabindex="-1"></a>    <span class="at">organism_id =</span> <span class="dv">9606</span>,</span>
<span id="cb14-46"><a href="#cb14-46" tabindex="-1"></a>    <span class="co"># tax ID can be found in function documentation or STRING database</span></span>
<span id="cb14-47"><a href="#cb14-47" tabindex="-1"></a>    <span class="at">binds_treatment =</span> binds_treatment,</span>
<span id="cb14-48"><a href="#cb14-48" tabindex="-1"></a>    <span class="at">plot =</span> <span class="cn">TRUE</span></span>
<span id="cb14-49"><a href="#cb14-49" tabindex="-1"></a>  )</span></code></pre></div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
