<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Dina Schuster" />

<meta name="date" content="2024-03-27" />

<title>Input Preparation Workflow</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Input Preparation Workflow</h1>
<h4 class="author">Dina Schuster</h4>
<h4 class="date">2024-03-27</h4>



<style>
body {
text-align: justify}
</style>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This vignette will give you an overview of how you can prepare the
quantitative protein/peptide matrix output from common search engines
and software such as <a href="https://biognosys.com/software/spectronaut/">Spectronaut</a>,
MaxQuant, <a href="https://www.thermofisher.com/order/catalog/product/OPTON-30945">Proteome
Discoverer</a> and <a href="https://skyline.ms/project/home/software/Skyline/begin.view">Skyline</a>
for the analysis with <strong>protti</strong>. Due to its modular and
flexible structure <strong>protti</strong> can be used on the output of
common bottom-up proteomics search engines irrespective of the
measurement mode (DDA, DIA, targeted-MS).</p>
<p>Furthermore, you are not only restricted to reports from the above
mentioned search engines. As long as your data has a tabular format
(data frame) and a specific minimal number of data columns you can
analyse it with <strong>protti</strong>. The columns minimally required
contain information on sample, condition, intensity, protein ID and the
level the intensity is based on (fragment, precursor, peptide) if
different from protein intensity. Depending on the analysis many more
columns can be useful, but they are not required. Ultimately, your data
should have a structure similar to this:</p>
<table>
<thead>
<tr class="header">
<th align="center">Sample</th>
<th align="center">Protein ID</th>
<th align="center">Peptide Sequence</th>
<th align="center">Condition</th>
<th align="center">Intensity</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">sample1</td>
<td align="center">P62942</td>
<td align="center">PEPTIDER</td>
<td align="center">treated</td>
<td align="center">14000</td>
</tr>
<tr class="even">
<td align="center">sample2</td>
<td align="center">P62942</td>
<td align="center">PEPTI</td>
<td align="center">treated</td>
<td align="center">15000</td>
</tr>
<tr class="odd">
<td align="center">sample3</td>
<td align="center">P62942</td>
<td align="center">PEPTIDE</td>
<td align="center">treated</td>
<td align="center">14500</td>
</tr>
<tr class="even">
<td align="center">sample4</td>
<td align="center">P62942</td>
<td align="center">PEPTIDER</td>
<td align="center">control</td>
<td align="center">18000</td>
</tr>
<tr class="odd">
<td align="center">sample5</td>
<td align="center">P62942</td>
<td align="center">PEPTI</td>
<td align="center">control</td>
<td align="center">21000</td>
</tr>
<tr class="even">
<td align="center">sample6</td>
<td align="center">P62942</td>
<td align="center">PEPTIDE</td>
<td align="center">control</td>
<td align="center">19000</td>
</tr>
</tbody>
</table>
<p>It is very important, that each unit of the level you perform your
analysis on (e.g. peptide) has a single unique intensity associated with
it. If, for example, a peptide has two different intensities,
<strong>protti</strong> would not know how to deal with this and many
functions will likely fail.</p>
<p>Data should always be organised in a format called <a href="https://r4ds.had.co.nz/tidy-data.html">tidy data</a>. That means
data should be contained in a long format (e.g. all sample names in one
column) rather than a wide format (e.g. each sample name in its own
column with intensity as the content of the columns). You can easily
achieve this by using the <code>pivot_longer()</code> function from the
<a href="https://tidyr.tidyverse.org/index.html"><code>tidyr</code></a>
package. The output of many search engines already contains tidy data
and working with it is very easy because you can refer to information
with only one variable. <strong>protti</strong> is designed to work
together well with the <a href="https://www.tidyverse.org"><code>tidyverse</code></a> package
family that is build around the concept of tidy data.</p>
<div id="protein-centric-analysis" class="section level2">
<h2>Protein-centric analysis</h2>
<p>Many search engines provide the user with protein intensities.
However, it is also possible to calculate protein intensities directly
from precursor intensities with the <strong>protti</strong> function
<code>calculate_protein_abundance()</code>. <strong>Protti</strong>
implements the <code>&quot;iq&quot;</code> method, previously implemented in the R
package <a href="https://academic.oup.com/bioinformatics/article/36/8/2611/5697917"><code>iq</code></a>
which performs protein quantification based on the maximal peptide ratio
extraction algorithm adapted from the MaxLFQ algorithm (<a href="https://www.sciencedirect.com/science/article/pii/S1535947620333107?via%3Dihub">Cox,
J. 2013</a>).</p>
<p>One advantage of calculating the protein abundance with
<strong>protti</strong> is the possibility to median normalise run
intensities on the precursor level. This is closer to the actually
acquired intensities and thus sample concentrations than if
normalisation is performed on the protein level. Some search engines
provide the option for automatic median normalisation but not all.
Furthermore, some search engines calculate protein intensities by
summation of precursor intensities irrespective of missingness of
peptides in certain samples. In these cases the maximal peptide ratio
implemented in extraction algorithm provides a more robust calculation
of protein intensities.</p>
<p>If you prefer to use protein intensities provided by the seach engine
of your choice this is not a problem and we will show how some of this
information can be converted into the right format.</p>
</div>
<div id="loading-packages" class="section level2">
<h2>Loading packages</h2>
<p>We will demonstrate how most outputs can be converted with functions
from the R packages <a href="https://magrittr.tidyverse.org/index.html"><code>magrittr</code></a>,
<a href="https://dplyr.tidyverse.org/index.html"><code>dplyr</code></a>,
<a href="https://tidyr.tidyverse.org/index.html"><code>tidyr</code></a>
and <a href="https://stringr.tidyverse.org/"><code>stringr</code></a>.
You can load packages after you installed them with the
<code>library()</code> function.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(stringr)</span></code></pre></div>
<p>Note that we are using the R package <a href="https://magrittr.tidyverse.org/index.html"><code>magrittr</code></a>
because of its pipe operator <code>%&gt;%</code>. It takes the output of
the preceding function and supplies it as the first argument of the
following function. Using <code>%&gt;%</code> makes code easier to read
and follow.</p>
</div>
</div>
<div id="spectronaut" class="section level1">
<h1>Spectronaut</h1>
<p>Spectronaut reports already contain data in the tidy data format.
Therefore nothing needs to be changed in order to use them with
<strong>protti</strong>. However, the columns we would recommend (not
all columns are required) to export from Spectronaut are:</p>
<ul>
<li>R.Condition (condition names)</li>
<li>R.FileName (file names)</li>
<li>PG.ProteinAccessions (protein identifiers)</li>
<li>PEP.IsProteotypic (logical indicating if peptide is
proteotypic)</li>
<li>PEP.StrippedSequence (peptide sequence)</li>
<li>PEP.NrOfMissedCleavages (number of missed cleavages, relevant for
quality control)</li>
<li>EG.IsDecoy (logical indicating if peptide is a decoy match)</li>
<li>EG.PrecursorId (peptide precursor ID)</li>
<li>FG.Quantity (precursor quantity, required for peptide-centric
analyses)</li>
<li>FG.Charge (precursor charge state, relevant for quality
control)</li>
<li>PG.Quantity (protein group quantity, required for protein-centric
analyses if you prefer not to calculate them with
<code>calculate_protein_abundance()</code>)</li>
</ul>
<p>Please make sure that the report is a .csv file. You can use the
<code>read_protti()</code> function in order to load the spectronaut
report into R. This function is a wrapper around the fast
<code>fread()</code> function from the <a href="https://rdatatable.gitlab.io/data.table/"><code>data.table</code></a>
package and the <code>clean_names()</code> function from the <a href="https://sfirke.github.io/janitor/"><code>janitor</code></a>
package. This will allow you to not only load your data into R very
fast, but also to clean up the column names into lower snake case. This
will make it easier to remember them and to use them in your data
analysis. For the Spectronaut columns <code>R.FileName</code> will
change for example into <code>r_file_name</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># To read in your own data you can use read_protti()</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>spectronaut_data <span class="ot">&lt;-</span> <span class="fu">read_protti</span>(<span class="at">filename =</span> <span class="st">&quot;mydata/spectronaut.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="maxquant" class="section level1">
<h1>MaxQuant</h1>
<p>Depending on which analysis you are performing you will have to use
different outputs. For peptide-centric analyses we would recommend to
use the <code>evidence.txt</code> file. If you want to perform a
protein-centric analysis and you want to use protein quantities
calculated by MaxQuant, you need the <code>proteinGroups.txt</code>
file. However, you can also apply the maximal peptide ratio extraction
algorithm from the <code>iq</code> R package implemented in the
<code>protein_abundance_calculation()</code> function of
<strong>protti</strong>. This allows you to only use the
<code>evidence.txt</code> file. The resulting protein intensities are
identical since they were calculated with the same algorithm.</p>
<div id="peptide-centric-analysislip-ms-analysis" class="section level2">
<h2>Peptide-centric analysis/LiP-MS analysis</h2>
<p>In case you are interested in performing a
<strong>peptide-centric</strong> analysis (necessary for LiP-MS), you
should use the <code>evidence.txt</code> file provided in the search
output of MaxQuant.</p>
<p>The <code>evidence.txt</code> file basically contains all the
information we need to run <strong>protti</strong>. It is also contained
in a long format which makes it easy to read in and use directly. One
thing to take into consideration is the lack of a column for information
on proteotypicity of peptides. However, this information can be inferred
from the <code>Proteins</code> column if it contains more than one
protein ID. You can extract this information and create a new column
called <code>is_proteotypic</code> containing logicals that will be
<code>TRUE</code> if the <code>Proteins</code> column does not contain a
semicolon and <code>FALSE</code> if it does (this indicates that the
peptide belongs to more than one protein). As mentioned in the data
analysis vignettes this information is necessary for the analysis of
LiP-MS data but it could be also considered for the correct calculation
of protein abundances.</p>
<p>Another column that is required for the analysis of your data is a
column indicating conditions to which certain samples belong. This can
be easily added to the evidence file by joining a data frame containing
the specific annotations. You can create such a data frame in Excel and
import it into R for a large number of samples or just create it
directly in R.</p>
<p>MaxQuant output provides information on decoy hits contained in the
column <code>reverse</code> and also has information on whether your hit
is a contaminant <code>potential_contaminant</code>. You should filter
these out before the analysis. However, the contaminant column can be
used for quality control.</p>
<p>One important thing for MaxQuant data is to <strong>make sure that
you only have one intensity assigned to each peptide or
precursor.</strong> You can do this by summing up all intensities that
MaxQuant exports (these can be MULTI-MSMS, MSMS, ISO-MSMS, MULTI-MATCH,
ISO-SECPEP) or you can filter for example for precursors with MULTI-MSMS
quantification and only use these.</p>
<p>In this section we will show you how to read in the file with
<code>read_protti()</code> and how to create the
<code>is_proteotypic</code> column and the <code>condition</code> column
(minimally required) with the help of the <code>stringr</code> and
<code>dplyr</code> packages. How to filter your data best is described
in the data analysis vignettes.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># To read in your own data you can use read_protti()</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">read_protti</span>(<span class="at">filename =</span> <span class="st">&quot;yourpath/evidence.txt&quot;</span>)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>evidence_proteotypic <span class="ot">&lt;-</span> evidence <span class="sc">%&gt;%</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="co"># adds new column with logicals that are TRUE if the peptide can be assigned</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="co"># to only one protein and FALSE if it can be assigned to multiple</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">is_proteotypic =</span> <span class="fu">str_detect</span>(</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>    <span class="at">string =</span> proteins,</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>    <span class="at">pattern =</span> <span class="st">&quot;;&quot;</span>,</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>    <span class="at">negate =</span> <span class="cn">TRUE</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>  <span class="co"># adds new column with logicals indicating if peptide is coming from a potential contaminant</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">is_contaminant =</span> <span class="fu">ifelse</span>(potential_contaminant <span class="sc">==</span> <span class="st">&quot;+&quot;</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>))</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="co"># Make an annotation data frame and merge it with your data frame to obtain conditions</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="co"># We are annotating sample 1-3 as controls and samples 4-6 as treated conditions</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="fu">c</span>( <span class="co"># make sure that the names are the same name as in your report</span></span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a>  <span class="st">&quot;sample1&quot;</span>,</span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a>  <span class="st">&quot;sample2&quot;</span>,</span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a>  <span class="st">&quot;sample3&quot;</span>,</span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a>  <span class="st">&quot;sample4&quot;</span>,</span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a>  <span class="st">&quot;sample5&quot;</span>,</span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a>  <span class="st">&quot;sample6&quot;</span></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a>)</span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a></span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a>condition <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span>,</span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span>,</span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span></span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a>)</span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a></span>
<span id="cb3-36"><a href="#cb3-36" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(file_name, condition)</span>
<span id="cb3-37"><a href="#cb3-37" tabindex="-1"></a></span>
<span id="cb3-38"><a href="#cb3-38" tabindex="-1"></a><span class="co"># Combine your long data frame with the annotation</span></span>
<span id="cb3-39"><a href="#cb3-39" tabindex="-1"></a>evidence_annotated <span class="ot">&lt;-</span> evidence_proteotypic <span class="sc">%&gt;%</span></span>
<span id="cb3-40"><a href="#cb3-40" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="at">y =</span> annotation, <span class="at">by =</span> <span class="st">&quot;file_name&quot;</span>)</span></code></pre></div>
</div>
<div id="protein-centric-analysis-1" class="section level2">
<h2>Protein-centric analysis</h2>
<p>For <strong>protein-centric</strong> analyses you can use the
<code>proteinGroups.txt</code> file provided by MaxQuant. This file
contains information in a wide format where each sample has its own
column containing intensity values. Therefore, we need to transform this
data into a long format to meet the conditions of tidy data.</p>
<p>We will filter the data and use <code>tidyr</code>’s
<code>pivot_longer()</code> to change the format to long format.
Furthermore, we produce an annotation data frame to create a
<code>conditions</code> column. The filtering is only done in order to
remove proteins with potentially low quality. Further filtering for
decoys and potential contaminants should be performed based on the data
analysis vignettes.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># To read in your own data you can use read_protti()</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>protein_groups <span class="ot">&lt;-</span> <span class="fu">read_protti</span>(<span class="at">filename =</span> <span class="st">&quot;yourpath/proteinGroups.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="co"># adds new column with logicals indicating if protein is a potential contaminant,</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="co"># you can filter these out later on. You should also consider filtering out proteins</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  <span class="co"># that were &quot;only identified by site&quot; and reverse hits, as well as proteins with only</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>  <span class="co"># one identified peptide</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">is_potential_contaminant =</span> <span class="fu">ifelse</span>(potential_contaminant <span class="sc">==</span> <span class="st">&quot;+&quot;</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>))</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co"># Change wide format to long format and create new columns called `r_file_name`and `intensity`</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>protein_groups_long <span class="ot">&lt;-</span> protein_groups <span class="sc">%&gt;%</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">&quot;intensity_&quot;</span>),</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;file_name&quot;</span>,</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">&quot;intensity&quot;</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>  )</span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="co"># Make an annotation data frame and merge it with your data frame to obtain conditions</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a><span class="co"># We are annotating sample 1-3 as controls and samples 4-6 as treated conditions</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="fu">c</span>( <span class="co"># make sure that the names are the same name as in your report</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a>  <span class="st">&quot;intensity_sample1&quot;</span>,</span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a>  <span class="st">&quot;intensity_sample2&quot;</span>,</span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a>  <span class="st">&quot;intensity_sample3&quot;</span>,</span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a>  <span class="st">&quot;intensity_sample4&quot;</span>,</span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>  <span class="st">&quot;intensity_sample5&quot;</span>,</span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a>  <span class="st">&quot;intensity_sample6&quot;</span></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a>)</span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a>condition <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb4-32"><a href="#cb4-32" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb4-33"><a href="#cb4-33" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span>,</span>
<span id="cb4-34"><a href="#cb4-34" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span>,</span>
<span id="cb4-35"><a href="#cb4-35" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span></span>
<span id="cb4-36"><a href="#cb4-36" tabindex="-1"></a>)</span>
<span id="cb4-37"><a href="#cb4-37" tabindex="-1"></a></span>
<span id="cb4-38"><a href="#cb4-38" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(file_name, condition)</span>
<span id="cb4-39"><a href="#cb4-39" tabindex="-1"></a></span>
<span id="cb4-40"><a href="#cb4-40" tabindex="-1"></a><span class="co"># Combine your long data frame with the annotation</span></span>
<span id="cb4-41"><a href="#cb4-41" tabindex="-1"></a>protein_groups_annotated <span class="ot">&lt;-</span> protein_groups_long <span class="sc">%&gt;%</span></span>
<span id="cb4-42"><a href="#cb4-42" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="at">y =</span> annotation, <span class="at">by =</span> <span class="st">&quot;file_name&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="skyline" class="section level1">
<h1>Skyline</h1>
<p>The Skyline output is already in long format, however, to process it
you need to sum up the transition intensities to obtain the intensity of
one precursor. If you prefer to analyse your data on the fragment level,
you should create a column that uniquely identifies each fragment of
each precursor. You could do that by pasting together the peptide
sequence with the charge and the product m/z.</p>
<p>The required Skyline output columns include:</p>
<ul>
<li>Peptide Sequence</li>
<li>Protein Name</li>
<li>Replicate Name</li>
<li>Precursor Charge</li>
<li>Product Mz (we are exporting these to distinguish transitions)</li>
<li>Area (or Intensity, depending on what you prefer to use)</li>
</ul>
<p>You can add replicate and condition annotations in Skyline directly.
However, we will explain in this section how you can also do it in R. If
you want to analyse your data on the protein abundance level you will
have to combine the precursor intensities to obtain one value for
protein abundance. This could be done using the
<code>calculate_protein_abundance()</code> function from
<strong>protti</strong>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>skyline_data <span class="ot">&lt;-</span> <span class="fu">read_protti</span>(<span class="at">filename =</span> <span class="st">&quot;yourpath/skyline.csv&quot;</span>)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>skyline_data_int <span class="ot">&lt;-</span> skyline_data <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  <span class="co"># create a column with precursor information</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">precursor =</span> <span class="fu">paste0</span>(peptide_sequence, <span class="st">&quot;_&quot;</span>, charge)) <span class="sc">%&gt;%</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>  <span class="fu">group_by</span>(replicate_name, precursor) <span class="sc">%&gt;%</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>  <span class="co"># making a new column containing the summed up intensities of all transitions of one precursor</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sum_intensity =</span> <span class="fu">sum</span>(area)) <span class="sc">%&gt;%</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(product_mz, area)) <span class="sc">%&gt;%</span> <span class="co"># removing the columns we don&#39;t need</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="co"># removing duplicated rows from the data frame</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="co"># Add annotation</span></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="co"># make sure that the names are the same name as in your report</span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>replicate_name <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a>  <span class="st">&quot;sample_1&quot;</span>,</span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>  <span class="st">&quot;sample_2&quot;</span>,</span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a>  <span class="st">&quot;sample_3&quot;</span>,</span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a>  <span class="st">&quot;sample_1&quot;</span>,</span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a>  <span class="st">&quot;sample_2&quot;</span>,</span>
<span id="cb5-21"><a href="#cb5-21" tabindex="-1"></a>  <span class="st">&quot;sample_3&quot;</span></span>
<span id="cb5-22"><a href="#cb5-22" tabindex="-1"></a>)</span>
<span id="cb5-23"><a href="#cb5-23" tabindex="-1"></a></span>
<span id="cb5-24"><a href="#cb5-24" tabindex="-1"></a>condition <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb5-25"><a href="#cb5-25" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb5-26"><a href="#cb5-26" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb5-27"><a href="#cb5-27" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb5-28"><a href="#cb5-28" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span>,</span>
<span id="cb5-29"><a href="#cb5-29" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span>,</span>
<span id="cb5-30"><a href="#cb5-30" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span></span>
<span id="cb5-31"><a href="#cb5-31" tabindex="-1"></a>)</span>
<span id="cb5-32"><a href="#cb5-32" tabindex="-1"></a></span>
<span id="cb5-33"><a href="#cb5-33" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(replicate_name, condition)</span>
<span id="cb5-34"><a href="#cb5-34" tabindex="-1"></a></span>
<span id="cb5-35"><a href="#cb5-35" tabindex="-1"></a><span class="co"># Combine your long data frame with the annotation</span></span>
<span id="cb5-36"><a href="#cb5-36" tabindex="-1"></a>skyline_annotated <span class="ot">&lt;-</span> skyline_data_int <span class="sc">%&gt;%</span></span>
<span id="cb5-37"><a href="#cb5-37" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="at">y =</span> annotation, <span class="at">by =</span> <span class="st">&quot;replicate_name&quot;</span>)</span></code></pre></div>
</div>
<div id="proteome-discoverer" class="section level1">
<h1>Proteome Discoverer</h1>
<p>The Proteome Discoverer output contains data in wide format (one
column for each sample). Similar to MaxQuant there is also the option
for a peptide or a protein-centric export. We will discuss both cases in
this segment.</p>
<div id="peptide-centric-analysislip-ms-analysis-1" class="section level2">
<h2>Peptide-centric analysis/LiP-MS analysis</h2>
<p>For a <strong>peptide-centric</strong> or a LiP-MS analysis please
export the “Peptide Groups” report. Before preparing your export you can
add the column “sequence” to your table otherwise Proteome Discoverer
will only export the “annotated sequence” column which includes the
preceding and following amino acids in the protein sequence.</p>
<p>The required columns include:</p>
<ul>
<li>Sequence</li>
<li>Modifications</li>
<li>Number Proteins</li>
<li>Contaminant</li>
<li>Master Protein Accessions</li>
<li>Abundance or normalized abundance columns</li>
<li>Quan Info</li>
</ul>
<p>After saving the report as an Excel file please convert it to a .csv
file, simply by opening it and saving it as such.</p>
<p>We will read in the file using <code>read_protti()</code> and then
select the columns we are interested in. You can use the
<code>contaminant</code> column for qualitiy control. The
<code>number_proteins</code> column contains information on the
proteotypicity. If this is 1 then the peptide is proteotypic. If you
want to analyse your data qualitatively only with quality control
functions of <strong>protti</strong> you can keep peptides without
quantifications. Before you start your quantitative analysis remove
observations that are labeled <code>&quot;No Quan Values&quot;</code> in the
<code>quan_info</code> column. In the below example they are filtered
out at this step, but you can keep them and only filter them out
later.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>pd_pep_data <span class="ot">&lt;-</span> <span class="fu">read_protti</span>(<span class="st">&quot;yourpath/PDpeptides.csv&quot;</span>)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co"># Select relevant columns</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>pd_pep_selected <span class="ot">&lt;-</span> pd_pep_data <span class="sc">%&gt;%</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>  <span class="fu">select</span>(</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>    sequence,</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>    modifications,</span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>    number_proteins,</span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>    contaminant,</span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>    master_protein_accessions,</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>    <span class="fu">starts_with</span>(<span class="st">&quot;abundances_grouped&quot;</span>), <span class="co"># select all columns that start with &quot;abundances_grouped&quot;</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a>    quan_info</span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a>  )</span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co"># Filter data frame</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a>pd_pep_filtered <span class="ot">&lt;-</span> pd_pep_selected <span class="sc">%&gt;%</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a>  <span class="fu">filter</span>(contaminant <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> <span class="co"># remove annotated contaminants</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a>  <span class="fu">filter</span>(number_proteins <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="co"># select proteotypic peptides</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a>  <span class="fu">filter</span>(quan_info <span class="sc">!=</span> <span class="st">&quot;No Quan Values&quot;</span>) <span class="co"># remove peptides that have no quantification values</span></span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a><span class="co"># Convert into long format</span></span>
<span id="cb6-23"><a href="#cb6-23" tabindex="-1"></a>pd_pep_long <span class="ot">&lt;-</span> pd_pep_filtered <span class="sc">%&gt;%</span></span>
<span id="cb6-24"><a href="#cb6-24" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb6-25"><a href="#cb6-25" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">&quot;abundances&quot;</span>),</span>
<span id="cb6-26"><a href="#cb6-26" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;file_name&quot;</span>,</span>
<span id="cb6-27"><a href="#cb6-27" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">&quot;intensity&quot;</span></span>
<span id="cb6-28"><a href="#cb6-28" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb6-29"><a href="#cb6-29" tabindex="-1"></a>  <span class="co"># combine peptide sequence and modifications to make a precursor column</span></span>
<span id="cb6-30"><a href="#cb6-30" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">precursor =</span> <span class="fu">paste</span>(sequence, modifications))</span>
<span id="cb6-31"><a href="#cb6-31" tabindex="-1"></a></span>
<span id="cb6-32"><a href="#cb6-32" tabindex="-1"></a><span class="co"># Make annotation data frame</span></span>
<span id="cb6-33"><a href="#cb6-33" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="fu">c</span>( <span class="co"># make sure that the names are the same name as in your report</span></span>
<span id="cb6-34"><a href="#cb6-34" tabindex="-1"></a>  <span class="st">&quot;abundances_grouped_f1&quot;</span>,</span>
<span id="cb6-35"><a href="#cb6-35" tabindex="-1"></a>  <span class="st">&quot;abundances_grouped_f2&quot;</span>,</span>
<span id="cb6-36"><a href="#cb6-36" tabindex="-1"></a>  <span class="st">&quot;abundances_grouped_f3&quot;</span>,</span>
<span id="cb6-37"><a href="#cb6-37" tabindex="-1"></a>  <span class="st">&quot;abundances_grouped_f4&quot;</span>,</span>
<span id="cb6-38"><a href="#cb6-38" tabindex="-1"></a>  <span class="st">&quot;abundances_grouped_f5&quot;</span>,</span>
<span id="cb6-39"><a href="#cb6-39" tabindex="-1"></a>  <span class="st">&quot;abundances_grouped_f6&quot;</span></span>
<span id="cb6-40"><a href="#cb6-40" tabindex="-1"></a>)</span>
<span id="cb6-41"><a href="#cb6-41" tabindex="-1"></a></span>
<span id="cb6-42"><a href="#cb6-42" tabindex="-1"></a>condition <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb6-43"><a href="#cb6-43" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb6-44"><a href="#cb6-44" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb6-45"><a href="#cb6-45" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb6-46"><a href="#cb6-46" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span>,</span>
<span id="cb6-47"><a href="#cb6-47" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span>,</span>
<span id="cb6-48"><a href="#cb6-48" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span></span>
<span id="cb6-49"><a href="#cb6-49" tabindex="-1"></a>)</span>
<span id="cb6-50"><a href="#cb6-50" tabindex="-1"></a></span>
<span id="cb6-51"><a href="#cb6-51" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(file_name, condition)</span>
<span id="cb6-52"><a href="#cb6-52" tabindex="-1"></a></span>
<span id="cb6-53"><a href="#cb6-53" tabindex="-1"></a><span class="co"># Combine your long data frame with the annotation</span></span>
<span id="cb6-54"><a href="#cb6-54" tabindex="-1"></a>pd_pep_long_annotated <span class="ot">&lt;-</span> pd_pep_long <span class="sc">%&gt;%</span></span>
<span id="cb6-55"><a href="#cb6-55" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="at">y =</span> annotation, <span class="at">by =</span> <span class="st">&quot;file_name&quot;</span>)</span></code></pre></div>
</div>
<div id="protein-centric-analysis-2" class="section level2">
<h2>Protein-centric analysis</h2>
<p>For a <strong>protein-centric</strong> or analysis please export the
“Proteins” report.</p>
<p>The required columns include:</p>
<ul>
<li>Accession</li>
<li>Description</li>
<li>Contaminant</li>
<li>Number Peptides</li>
<li>Abundance or normalized abundance columns</li>
</ul>
<p>After saving the report as an Excel file please convert it to a .csv
file, simply by opening it and saving it as such.</p>
<p>We will read in the file using <code>read_protti()</code> and then
select the columns we are interested in. Similar to above you can either
filter the <code>contaminant</code> and <code>number_peptides</code>
columns now or later.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>pd_prot_data <span class="ot">&lt;-</span> <span class="fu">read_protti</span>(<span class="st">&quot;yourpath/PDproteins.csv&quot;</span>)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co"># Select relevant columns</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>pd_prot_selected <span class="ot">&lt;-</span> pd_prot_data <span class="sc">%&gt;%</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>  <span class="fu">select</span>(</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>    accession,</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>    description,</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>    contaminant,</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>    number_peptides,</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>    <span class="fu">starts_with</span>(<span class="st">&quot;abundances_grouped&quot;</span>), <span class="co"># select all columns that start with &quot;abundances_grouped&quot;</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>  )</span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a><span class="co"># Filter data frame</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>pd_prot_data_filtered <span class="ot">&lt;-</span> pd_prot_selected <span class="sc">%&gt;%</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a>  <span class="fu">filter</span>(contaminant <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> <span class="co"># remove annotated contaminants</span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a>  <span class="fu">filter</span>(number_peptides <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="co"># select proteins with more than one identified peptide</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a><span class="co"># Convert into long format</span></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a>pd_prot_long <span class="ot">&lt;-</span> pd_prot_data_filtered <span class="sc">%&gt;%</span></span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">&quot;abundances&quot;</span>),</span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;file_name&quot;</span>,</span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">&quot;intensity&quot;</span></span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a>  )</span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a></span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a><span class="co"># Make annotation data frame</span></span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="fu">c</span>( <span class="co"># make sure that the names are the same name as in your report</span></span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a>  <span class="st">&quot;abundances_grouped_f1&quot;</span>,</span>
<span id="cb7-30"><a href="#cb7-30" tabindex="-1"></a>  <span class="st">&quot;abundances_grouped_f2&quot;</span>,</span>
<span id="cb7-31"><a href="#cb7-31" tabindex="-1"></a>  <span class="st">&quot;abundances_grouped_f3&quot;</span>,</span>
<span id="cb7-32"><a href="#cb7-32" tabindex="-1"></a>  <span class="st">&quot;abundances_grouped_f4&quot;</span>,</span>
<span id="cb7-33"><a href="#cb7-33" tabindex="-1"></a>  <span class="st">&quot;abundances_grouped_f5&quot;</span>,</span>
<span id="cb7-34"><a href="#cb7-34" tabindex="-1"></a>  <span class="st">&quot;abundances_grouped_f6&quot;</span></span>
<span id="cb7-35"><a href="#cb7-35" tabindex="-1"></a>)</span>
<span id="cb7-36"><a href="#cb7-36" tabindex="-1"></a></span>
<span id="cb7-37"><a href="#cb7-37" tabindex="-1"></a>condition <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb7-38"><a href="#cb7-38" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb7-39"><a href="#cb7-39" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb7-40"><a href="#cb7-40" tabindex="-1"></a>  <span class="st">&quot;control&quot;</span>,</span>
<span id="cb7-41"><a href="#cb7-41" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span>,</span>
<span id="cb7-42"><a href="#cb7-42" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span>,</span>
<span id="cb7-43"><a href="#cb7-43" tabindex="-1"></a>  <span class="st">&quot;treated&quot;</span></span>
<span id="cb7-44"><a href="#cb7-44" tabindex="-1"></a>)</span>
<span id="cb7-45"><a href="#cb7-45" tabindex="-1"></a></span>
<span id="cb7-46"><a href="#cb7-46" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(file_name, condition)</span>
<span id="cb7-47"><a href="#cb7-47" tabindex="-1"></a></span>
<span id="cb7-48"><a href="#cb7-48" tabindex="-1"></a><span class="co"># Combine your long data frame with the annotation</span></span>
<span id="cb7-49"><a href="#cb7-49" tabindex="-1"></a>pd_prot_long_annotated <span class="ot">&lt;-</span> pd_prot_long <span class="sc">%&gt;%</span></span>
<span id="cb7-50"><a href="#cb7-50" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="at">y =</span> annotation, <span class="at">by =</span> <span class="st">&quot;file_name&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="other-search-engines-and-software" class="section level1">
<h1>Other search engines and software</h1>
<p>As mentioned in the beginning of this vignette you can use the output
of any search engine as long as it contains the minimally required
columns. If it is not in the right format you can see if some of the
above transformations can be applied to your data. It is also always
useful to check if you can find additional columns that help you in your
analysis and that you can export from your search engine. Always make
sure that all of your observations are ones you are interested in. Check
if there are decoys, contaminants or non-proteotypic peptides in your
data. For protein-centric analysis, potentially remove quantifications
that rely on only a few peptides.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
