<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Dina Schuster" />

<meta name="date" content="2024-10-22" />

<title>Qualtiy Control (QC) Workflow</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Qualtiy Control (QC) Workflow</h1>
<h4 class="author">Dina Schuster</h4>
<h4 class="date">2024-10-22</h4>



<style>
body {
text-align: justify}
</style>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This vignette will give you an overview of how you can perform
quality control of your bottom-up proteomics or LiP-MS data. For help
with data input and to make sure that you are using the right format
please check out the <a href="https://jpquast.github.io/protti/articles/input_preparation_workflow.html">input
preparation vignette</a>. For help with data analysis you can check out
our <a href="https://jpquast.github.io/protti/articles/data_analysis_single_dose_treatment_workflow.html">single
dose treatment vignette</a> and our <a href="https://jpquast.github.io/protti/articles/data_analysis_dose_response_workflow.html">data
analysis vignette for dose-response data</a>.</p>
<p>When working with proteomics and limited proteolysis-coupled mass
spectrometry (<a href="https://doi.org/10.1038/nprot.2017.100">LiP-MS</a>) data, it is
important to ensure that the data is of sufficient quality before
inferring biological meaning from it. The goal is to assess if the data
quality is homogenous among the different measurements (i.e. samples) of
an experiment and if there are outliers that should be excluded.</p>
<p>In general, there are several ways in which this can be done:</p>
<ul>
<li>Inspection of .raw files acquired directly on the mass spectrometer
(e.g. using vendor specific applications to visualize mass spectrometry
data, such as the Thermo Scientific applications XCalibur™ and
FreeStyle™)</li>
<li>Different search engines and software offer quality control
functionalities (e.g. <a href="https://biognosys.com/software/spectronaut/">Spectronaut™</a>
offers a lot of QC options)</li>
<li>R packages (e.g. <a href="https://pubmed.ncbi.nlm.nih.gov/26653327/">PTXQC</a> which runs on
MaxQuant output or <a href="https://pubmed.ncbi.nlm.nih.gov/29978702/">rawDiag</a> which can
be used directly on raw files)</li>
</ul>
<p><em>Please note that this is not a complete list, it is just intended
to give you an idea of the different possibilities. There are many more
software options and R packages for quality control published to
date.</em></p>
</div>
<div id="protti" class="section level1">
<h1>protti</h1>
<p><strong>protti</strong> includes several functions that make it easy
for the user to compare the different samples in one experiment. Some
functions in this R package are specific for LiP-MS data analyses,
whereas others can also be used for general proteomics analyses.</p>
<p>You can read more about specific functions and how to use them by
calling e.g. <code>?qc_cvs</code> (for the <code>qc_cvs()</code>
function). Calling <code>?</code> followed by the function name will
display the function documentation and give you more detailed
information about the function. This can be done for any of the
functions included in the package.</p>
<p>This document will give you an overview of QC functions included in
<strong>protti</strong> and will show you how they can be applied to
your data. The examples in this file are run on synthetic data, which is
created with the function <code>create_synthetic_data()</code>.</p>
<p>The functions in <strong>protti</strong> are tailored towards the
analysis of DIA data, based on the output of the search engine <a href="https://biognosys.com/software/spectronaut/">Spectronaut™</a>.
However, if you have any other data, such as DDA data that was searched
with a different search engine, you can still apply
<strong>protti</strong>’s functions. Just make sure that your data frame
contains <a href="https://r4ds.had.co.nz/tidy-data.html">tidy data</a>.
That means data should be contained in a long format (e.g. all sample
names in one column) rather than a wide format (e.g. each sample name in
its own column). You can easily achieve this by using the
<code>pivot_longer()</code> function from the <code>tidyr</code>
package. If you are unsure about what your input data should look like,
please use the <code>create_synthetic_data()</code> function and compare
this to your data. You can read more about all of this <a href="https://jpquast.github.io/protti/articles/input_preparation_workflow.html">here</a>.</p>
<p>The input data should have a similar structure to this example:</p>
<table>
<colgroup>
<col width="12%" />
<col width="16%" />
<col width="28%" />
<col width="16%" />
<col width="25%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Sample</th>
<th align="center">Replicate</th>
<th align="center">Peptide Sequence</th>
<th align="center">Condition</th>
<th align="center">log2(Intensity)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">sample1</td>
<td align="center">1</td>
<td align="center">PEPTIDER</td>
<td align="center">treated</td>
<td align="center">14</td>
</tr>
<tr class="even">
<td align="center">sample1</td>
<td align="center">1</td>
<td align="center">PEPTI</td>
<td align="center">treated</td>
<td align="center">16</td>
</tr>
<tr class="odd">
<td align="center">sample1</td>
<td align="center">1</td>
<td align="center">PEPTIDE</td>
<td align="center">treated</td>
<td align="center">17</td>
</tr>
<tr class="even">
<td align="center">sample2</td>
<td align="center">1</td>
<td align="center">PEPTIDER</td>
<td align="center">untreated</td>
<td align="center">15</td>
</tr>
<tr class="odd">
<td align="center">sample2</td>
<td align="center">1</td>
<td align="center">PEPTI</td>
<td align="center">untreated</td>
<td align="center">18</td>
</tr>
<tr class="even">
<td align="center">sample2</td>
<td align="center">1</td>
<td align="center">PEPTIDE</td>
<td align="center">untreated</td>
<td align="center">12</td>
</tr>
</tbody>
</table>
<div id="getting-started" class="section level2">
<h2>Getting started</h2>
<p>Before we can start analysing our data, we need to load the
<strong>protti</strong> package. This is done by using the base R
function <code>library()</code>. In addition, we are also loading the
packages <code>magrittr</code> and <code>dplyr</code>. Both
<code>magrittr</code> and <code>dplyr</code> are part of the <a href="https://www.tidyverse.org">tidyverse</a>, a collection of packages
that provide useful functionalities for data processing and
visualisation. If you use many tidyverse packages in your workflow you
can easily load all at once by calling
<code>library(tidyverse)</code>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(protti)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<p>After having loaded the required packages we can create a synthetic
dataset, that contains data similar to data obtained from a treatment
experiment with e.g. a protein, metabolite or small-molecule. You can
skip this step if you have your own data ready.</p>
<p>We are creating a random dataset with 100 different proteins, out of
which 5 % are significantly changing upon treatment. The dataset
includes 3 replicates for 2 different conditions (treated and
untreated).</p>
<p>The output of the <code>create_synthetic_data()</code> function is
modeled after real LiP-MS data and its commonly observed data
distributions.</p>
<p>Please note that generally, quality control should be conducted on
raw unfiltered data (the direct output of your search engine of
choice).</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># by setting the seed we are making sure that the random object generation can be reproduced</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">create_synthetic_data</span>(</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  <span class="at">n_proteins =</span> <span class="dv">100</span>,</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  <span class="at">frac_change =</span> <span class="fl">0.05</span>,</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>  <span class="at">n_replicates =</span> <span class="dv">3</span>,</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>  <span class="at">n_conditions =</span> <span class="dv">2</span>,</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;effect_random&quot;</span>,</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>  <span class="at">additional_metadata =</span> <span class="cn">TRUE</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="quality-control" class="section level2">
<h2>Quality control</h2>
<div id="coefficients-of-variation" class="section level3">
<h3>Coefficients of variation</h3>
<p>A good first step of quality control is to check if coefficients of
variation between replicates are in a reasonable range. Ideally, these
should be below 15-20 %. If you see groups with higher CVs, there might
have been a sample preparation error or the instrument performance was
not ideal at the time of measurement. You can check if after
<code>normalise()</code> (depending on whether your search engine has
performed normalisation already) your CVs improve.</p>
<p>To conduct the CV-based quality control, we make use of the function
<code>qc_cvs()</code>. This function calculates the coefficients of
variation. The output can either be a table or a plot of coefficients of
variations for each condition. Below we are returning a CV table and a
violin plot, showing the CVs for different conditions.</p>
<p>The CVs (in percent) are calculated with the following formula:</p>
<div class="blackbox">
<p>
<span class="math display"><span class="math display">\[CV =
\frac{standard ~ deviation}{mean} *
100\]</span></span>
</p>
</div>
<p>Since the CV function works only with raw values, we will
backtransform our log2 transformed data and create a new column called
<code>raw_intensity</code> from the
<code>peptide_intensity_missing</code> column which contains peptide
intensities. To create the new column, we are using the function
<code>mutate()</code> from the R package <code>dplyr</code>. We also
make use of the pipe operator <code>%&gt;%</code> included in the R
package <code>magrittr</code> which takes the output of the preceding
function and supplies it as the first argument of the following
function. Using <code>%&gt;%</code> makes code easier to read and
follow.</p>
<p>The “combined” group of CVs contains CVs across all samples and not
only across the replicates of a certain condition. Ideally the combined
CVs should always be the group with the highest CV. This would indicate
that CVs within conditions are low and the treatment had and effect that
causes an increase in overall CVs. If the treatment did not have an
effect or only a very small one then the combined CV could be as high as
the CVs of individual conditions. If an individual condition has a
higher CV than the combined CV then there was potentially a problem with
one or multiple of the samples. You can check if there are suspicious
samples behaving differently from the rest in the other quality control
metrics. If there is a consistent different behaviour these samples
should potentially be excluded for analysis. Sometimes a condition can
also just be noisy because of the treatment in this condition that
causes a larger variation. Therefore comparing individual CVs with the
combined CV should only be done with caution.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>input <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  <span class="co"># as the data is log2 transformed, we need to transform it back before calculating the CVs</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">raw_intensity =</span> <span class="dv">2</span><span class="sc">^</span>peptide_intensity_missing)</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="fu">qc_cvs</span>(</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="at">grouping =</span> peptide,</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  <span class="at">condition =</span> condition,</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>  <span class="at">intensity =</span> raw_intensity,</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">FALSE</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>)</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="fu">qc_cvs</span>(</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>  <span class="at">grouping =</span> peptide,</span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>  <span class="at">condition =</span> condition,</span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a>  <span class="at">intensity =</span> raw_intensity,</span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a>  <span class="at">plot_style =</span> <span class="st">&quot;violin&quot;</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="number-of-identifications-ids" class="section level3">
<h3>Number of identifications (IDs)</h3>
<p>The number of protein or peptide identifications should be similar
for different samples. If there are significanlty less observations in
one sample, this might indicate a sample preparation or measurement
error.</p>
<p>For the analysis of the number of identifications of precursors,
peptides or proteins we use the function <code>qc_ids()</code>. This
function can return either a table or a plot. The output of this
function - and also of a lot of other <strong>protti</strong> functions
- can be plotted in an interactive plot that makes use of the R package
<code>plotly</code>. You can plot an interactive version of the plot by
setting <code>interactive = TRUE</code> within the function call.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">qc_ids</span>(</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="at">grouping =</span> protein,</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  <span class="at">intensity =</span> peptide_intensity_missing,</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>  <span class="at">condition =</span> condition,</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">FALSE</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>)</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="fu">qc_ids</span>(</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>  <span class="at">grouping =</span> protein,</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>  <span class="at">intensity =</span> peptide_intensity_missing,</span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>  <span class="at">condition =</span> condition,</span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a>  <span class="at">title =</span> <span class="st">&quot;Protein identifications per sample&quot;</span>,</span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">TRUE</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="peptide-types" class="section level3">
<h3>Peptide types</h3>
<p>Now that we had a look at protein IDs, we should also check peptide
identifications for homogeneity across samples. An important factor to
take into account when analysing LiP-MS data are peptide types (tryptic,
semi-tryptic, non-tryptic). They can give us an idea about the
reproducibility of cleavage events in the experiment and if our
treatment influenced protease activity or the digest in general.</p>
<p>In a standard LiP-MS experiment we use 3 different proteases:</p>
<ul>
<li>Proteinase K (unspecific, cleaves accessible protein regions)</li>
<li>Trypsin (specific, cleaves after lysines or arginines)</li>
<li>Lys-C (specific, cleaves after lysines)</li>
</ul>
<p>Peptide types can be assessed in Spectronaut directly or by using the
<strong>protti</strong> function <code>assign_peptide_type()</code>.
This function takes amino acids before and after the first and last
amino acid in the peptide into account. A peptide that starts after a
lysine or arginine and ends with a lysine or arginine is a
<em>tryptic</em> peptide. A peptide that has one proteinase K cleavage
site (not lysine or arginine) and one trypsin cleavage site either at
the beginning or at the end of the peptide is a <em>semi-tryptic</em>
peptide. A peptide that does not have any trypsin cleavage sites is a
<em>non-tryptic</em> peptide. If information about the amino acid before
and the last amino acid of a peptide are not available in your data you
can use the <code>find_peptide</code> function from the
<strong>protti</strong> package to create it.</p>
<p>We are going to use the function <code>qc_peptide_type()</code> to
evaluate the distribution of peptide types between our samples. The
function can return a plot or a table, based on either peptide type
counts or intensities. Both are equally important when assessing how
homogenous the samples are. For <code>method = intensity</code> the
function uses raw intensity values, so we are going to use the column
<code>raw_intensity</code> which we added previously.</p>
<p>In a typical LiP-MS experiment, there should be few non-tryptic
peptides. The peptide type distribution for fully- and semi-, and
non-tryptic peptides depends on your digestion conditions
(i.e. temperature, duration, concentration).</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">qc_peptide_type</span>(</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="at">peptide =</span> peptide,</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  <span class="at">pep_type =</span> pep_type,</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;intensity&quot;</span>,</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>  <span class="at">intensity =</span> raw_intensity,</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>  <span class="at">interactive =</span> <span class="cn">FALSE</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>)</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="fu">qc_peptide_type</span>(</span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>  <span class="at">peptide =</span> peptide,</span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a>  <span class="at">pep_type =</span> pep_type,</span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;count&quot;</span>,</span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a>  <span class="at">interactive =</span> <span class="cn">FALSE</span></span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="run-intensities" class="section level3">
<h3>Run intensities</h3>
<p>The function <code>qc_intensity_distribution()</code> plots all
precursor, peptide or protein intensities for each sample as a boxplot
if method “boxplot” is selected. This is helpful for quick assessment of
any major sample losses or measurement issues.</p>
<p>Run intensities can also be assessed by plotting the median run
intensities as a line plot. This helps you quickly assess if there are
any trends in your data. The function to use for this analysis is
<code>qc_median_intensities()</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">qc_intensity_distribution</span>(</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>  <span class="at">grouping =</span> peptide,</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>  <span class="at">intensity_log2 =</span> peptide_intensity_missing,</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>  <span class="at">plot_style =</span> <span class="st">&quot;boxplot&quot;</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>)</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="fu">qc_median_intensities</span>(</span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>  <span class="at">grouping =</span> peptide,</span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a>  <span class="at">intensity =</span> peptide_intensity_missing</span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="charge-states" class="section level3">
<h3>Charge states</h3>
<p>The charge state distibution of the detected peptides can be assessed
with <code>qc_charge_states()</code>. Similar to the peptide types, this
should also be homogenous for the different samples. The function can
return either a plot or a table based on counts of peptides with a
specific charge state or the intensities of these peptides. For
<code>method = &quot;intensity&quot;</code> the function requires the raw
intensity values created previously as its input.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">qc_charge_states</span>(</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  <span class="at">grouping =</span> peptide,</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>  <span class="at">charge_states =</span> charge,</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;intensity&quot;</span>,</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>  <span class="at">intensity =</span> raw_intensity,</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">TRUE</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="missed-cleavages" class="section level3">
<h3>Missed cleavages</h3>
<p>An important measure of efficient digests with e.g. trypsin are the
number of missed cleavages. The proteases trypsin and Lys-C both cleave
after positively charged amino acids lysine (K) and arginine (R). If a
measured peptide includes either K or R within its sequence, i.e. not at
the last position, it contains a missed cleavage site. These peptides
are important to take into account because they can tell you if your
tryptic digest was inefficient. The number of missed cleavages should
generally be low in a proteomics or LiP-MS dataset.</p>
<p>We are going to check the numbers of missed cleavages in our dataset
by using the function <code>qc_missed_cleavages()</code>. It can assess
missed cleavages based on the count of peptides with missed cleavages or
the intensities of the corresponding peptides. For
<code>method = &quot;intensity&quot;</code> the function uses the raw (not log2
transformed) intensity values. You can have the function either return a
plot or a table.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">qc_missed_cleavages</span>(</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>  <span class="at">grouping =</span> peptide,</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>  <span class="at">missed_cleavages =</span> n_missed_cleavage,</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;intensity&quot;</span>,</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>  <span class="at">intensity =</span> raw_intensity,</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">TRUE</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="sequence-coverage" class="section level3">
<h3>Sequence coverage</h3>
<p>The following function gives you insight into the protein coverage
(i.e. what percentage of the protein sequence is covered by the
identified peptides) in the dataset. For a LiP-MS experiment, or any
other peptide-centric proteomics experiment, high sequence coverages are
desirable.</p>
<p>To assess the protein coverage distribution, we are going to use the
function <code>qc_protein_coverage()</code>. If you do not have a column
containing the protein coverages in your data, you can use the function
<code>calculate_sequence_coverage()</code> to obtain this
information.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">qc_sequence_coverage</span>(</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="at">protein_identifier =</span> protein,</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="at">coverage =</span> coverage</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="peak-width" class="section level3">
<h3>Peak width</h3>
<p>In order to identify potential chromatographic issues that might have
occurred during the measurement we can have a look at the
chromatographic median peak widths over the complete chromatogram. To do
this we are going to use the function <code>qc_peak_width()</code>. This
function requires either the peak start and end times or the retention
time and peak width. This information can be obtained from Spectronaut
(or any other search engine of your choice). The peak widths should be
similar for all the measured samples of the experiment.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">qc_peak_width</span>(</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>  <span class="at">intensity =</span> peptide_intensity_missing,</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>  <span class="at">retention_time =</span> retention_time,</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>  <span class="at">peak_width =</span> peak_width</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="data-completeness" class="section level3">
<h3>Data completeness</h3>
<p>The function <code>qc_data_completeness()</code> checks how many of
all detected precursors, peptides or proteins were identified in each
sample. The function can return either a plot or a table.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">qc_data_completeness</span>(</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>  <span class="at">grouping =</span> peptide,</span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>  <span class="at">intensity =</span> peptide_intensity_missing,</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">TRUE</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="log2-intensity-distribution" class="section level3">
<h3>Log2 Intensity distribution</h3>
<p>For different kinds of analyses (e.g. t-tests) it is important that
your data intensity follows a normal distribution. To ensure that this
is the case, we are going to use the function
<code>qc_intensity_distriubution()</code>. The function returns a
histogram plot when <code>plot_style = &quot;histogram&quot;</code> showing how
the intensities are distributed.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">qc_intensity_distribution</span>(</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>  <span class="at">grouping =</span> peptide,</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>  <span class="at">intensity_log2 =</span> peptide_intensity_missing,</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>  <span class="at">plot_style =</span> <span class="st">&quot;histogram&quot;</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="sample-correlation" class="section level3">
<h3>Sample correlation</h3>
<p>Another approach to quality control is to check the correlation of
your samples. Ideally, replicates should cluster together and different
treatment conditions should be separated. We are now going to check if
this is the case for our data by using the function
<code>qc_sample_correlation()</code>. The function will return a
correlation heatmap with a comparison of all samples.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">qc_sample_correlation</span>(</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>  <span class="at">data =</span> input,</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>  <span class="at">grouping =</span> peptide,</span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>  <span class="at">intensity_log2 =</span> peptide_intensity_missing,</span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>  <span class="at">condition =</span> condition,</span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>  <span class="at">interactive =</span> <span class="cn">FALSE</span></span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="principal-component-analysis-pca" class="section level3">
<h3>Principal component analysis (PCA)</h3>
<p>Another popular quality control method (which could also be
considered part of data analysis) is the principal component analysis
(PCA). Generally, PCA is a method that reduces dimensionality of large
datasets. In our case this helps us to quickly assess how similar or
different our replicates and conditions are. We are going to use
<code>qc_pca()</code> to compute and plot a PCA for our data. Before
plotting your PCA you can check how much of your variance can be
explained with each of the principle components. To do that you can plot
a scree-plot using the function <code>qc_pca()</code> and setting
<code>plot_style = &quot;scree&quot;</code>. In this example we can see that the
variance can be best explained with PC1 and PC2 so we can go ahead and
plot these.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">qc_pca</span>(</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>  <span class="at">data =</span> data,</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  <span class="at">grouping =</span> peptide,</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>  <span class="at">intensity =</span> peptide_intensity_missing,</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>  <span class="at">condition =</span> condition,</span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>  <span class="at">digestion =</span> <span class="cn">NULL</span>,</span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a>  <span class="at">plot_style =</span> <span class="st">&quot;scree&quot;</span></span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a>)</span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a><span class="fu">qc_pca</span>(</span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a>  <span class="at">data =</span> data,</span>
<span id="cb14-13"><a href="#cb14-13" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb14-14"><a href="#cb14-14" tabindex="-1"></a>  <span class="at">grouping =</span> peptide,</span>
<span id="cb14-15"><a href="#cb14-15" tabindex="-1"></a>  <span class="at">intensity =</span> peptide_intensity_missing,</span>
<span id="cb14-16"><a href="#cb14-16" tabindex="-1"></a>  <span class="at">condition =</span> condition,</span>
<span id="cb14-17"><a href="#cb14-17" tabindex="-1"></a>  <span class="at">components =</span> <span class="fu">c</span>(<span class="st">&quot;PC1&quot;</span>, <span class="st">&quot;PC2&quot;</span>),</span>
<span id="cb14-18"><a href="#cb14-18" tabindex="-1"></a>  <span class="at">plot_style =</span> <span class="st">&quot;pca&quot;</span></span>
<span id="cb14-19"><a href="#cb14-19" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="ranked-intensity-distribution" class="section level3">
<h3>Ranked intensity distribution</h3>
<p>It can be useful to check which precursors, peptides or proteins are
the most or least abundant in the sample by ranking their intensities.
You can create a data frame containing ranked median intensities for the
whole experiment or ranked intensities for each sample using the
<code>qc_ranked_intensities()</code> function. In addition you can also
create a corresponding plot by setting the <code>plot</code> argument to
<code>TRUE</code>. If you would like to axis to not be log10 transformed
but rather log2 you can change the <code>y_axis_transformation</code>
argument.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="co"># Plot ranked peptide intensities</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a><span class="fu">qc_ranked_intensities</span>(</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>  <span class="at">data =</span> data,</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>  <span class="at">sample =</span> sample,</span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>  <span class="at">grouping =</span> peptide,</span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>  <span class="at">intensity_log2 =</span> peptide_intensity,</span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="additional-qc-functions" class="section level3">
<h3>Additional QC functions</h3>
<p>There are two additional QC functions that are not included in this
vignette:</p>
<ul>
<li><code>qc_contaminants()</code> is a function that can be used if
proteins are assigned as contaminants. This information can for example
be exported from MaxQuant. The function calculates the percentage of
proteins annotated to be contaminants and returns either a table or a
plot.</li>
<li><code>qc_proteome_coverage()</code> calculates the proteome coverage
for each sample and the whole experiment and returns either a plot or a
table.</li>
</ul>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
